<?php

	include '../../../../common/include/common.php';
    include_once '../../../config.php';
    $module = modules::GATK;
    $name = docSN::toolDocs;

    printHeader($module, $name, topSN::guide);

    $selected_major_version = 4;
    $dirs = lookupVersionDirs($selected_major_version);
?>

<link type='text/css' rel='stylesheet' href='gatkDoc.css'>

<div class="row hide_me_html" style="margin: 14px 3px 14px 3px;" id="tippy-top">

	<i class='<?php print ico::toolDocsIcon; ?>'></i> <em>Showing tool doc from version 4.3.0.0 | The latest version is <?php print $latest_version; ?></em>
	<div class="btn-group pull-right">
		<a class="btn btn-warning dropdown-toggle" data-toggle="dropdown" href="#">
			4.3.0.0
			<span class="caret"></span>
		</a>
		<ul class="dropdown-menu">
			<?php foreach($dirs as $dir) { ?>
			<li class="hide_me_html"><a tabindex='-1' href='../<?=$dir?>' ><?=$dir?></a></li>
			<?php } ?>
		</ul>
	</div>
	<hr />

</div>


<div class='row' id="top">



	<?php $group = 'Flow Based Tools'; ?>

	<section class="col-md-4">
		<aside class="well">
			<a class="hide_me_php" href="index.html"><h4><i class='fa fa-chevron-left'></i> Back to Tool Documentation Index</h4></a>
			<a class="hide_me_html" href=".."><h4><i class='fa fa-chevron-left'></i> Back to Tool Documentation Index</h4></a>
		</aside>
		<aside class="well hide_me_html">
			<h3>Browse tools by category</h3>

        <style>
            #sidenav .accordion-body a {
                color : gray;
            }

            .accordion-body li {
                list-style : none;
            }
        </style>

        <ul class="nav nav-pills nav-stacked" id="sidenav">
        		<hr>
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#CopyNumberVariantDiscovery">Copy Number Variant Discovery</a>
							<div id="CopyNumberVariantDiscovery"
								<?php echo ($group == 'Copy Number Variant Discovery')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="org_broadinstitute_hellbender_tools_copynumber_AnnotateIntervals.html">AnnotateIntervals</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_copynumber_CallCopyRatioSegments.html">CallCopyRatioSegments</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_copynumber_CreateReadCountPanelOfNormals.html">CreateReadCountPanelOfNormals</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_copynumber_DenoiseReadCounts.html">DenoiseReadCounts</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_copynumber_DetermineGermlineContigPloidy.html">DetermineGermlineContigPloidy</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_copynumber_FilterIntervals.html">FilterIntervals</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_copynumber_GermlineCNVCaller.html">GermlineCNVCaller</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_copynumber_ModelSegments.html">ModelSegments</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_copynumber_plotting_PlotDenoisedCopyRatios.html">PlotDenoisedCopyRatios</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_copynumber_plotting_PlotModeledSegments.html">PlotModeledSegments</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_copynumber_PostprocessGermlineCNVCalls.html">PostprocessGermlineCNVCalls</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#CoverageAnalysis">Coverage Analysis</a>
							<div id="CoverageAnalysis"
								<?php echo ($group == 'Coverage Analysis')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_rnaseq_ASEReadCounter.html">ASEReadCounter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_AnalyzeSaturationMutagenesis.html">AnalyzeSaturationMutagenesis</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_copynumber_CollectAllelicCounts.html">CollectAllelicCounts</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_copynumber_CollectReadCounts.html">CollectReadCounts</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_CountBases.html">CountBases</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_pipelines_CountBasesSpark.html">CountBasesSpark</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_CountReads.html">CountReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_pipelines_CountReadsSpark.html">CountReadsSpark</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_coverage_DepthOfCoverage.html">DepthOfCoverage</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_rnaseq_GeneExpressionEvaluation.html">GeneExpressionEvaluation</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_mutect_GetNormalArtifactData.html">GetNormalArtifactData</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_contamination_GetPileupSummaries.html">GetPileupSummaries</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_LocalAssembler.html">LocalAssembler</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_qc_Pileup.html">Pileup</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_PileupSpark.html">PileupSpark</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#DiagnosticsandQualityControl">Diagnostics and Quality Control</a>
							<div id="DiagnosticsandQualityControl"
								<?php echo ($group == 'Diagnostics and Quality Control')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_util_AccumulateQualityYieldMetrics.html">AccumulateQualityYieldMetrics (Picard)</a>
											</li>
											<li>
												<a href="picard_vcf_AccumulateVariantCallingMetrics.html">AccumulateVariantCallingMetrics (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_bqsr_AnalyzeCovariates.html">AnalyzeCovariates</a>
											</li>
											<li>
												<a href="picard_sam_BamIndexStats.html">BamIndexStats (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_sv_evidence_CalcMetadataSpark.html">CalcMetadataSpark</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_contamination_CalculateContamination.html">CalculateContamination</a>
											</li>
											<li>
												<a href="picard_fingerprint_CalculateFingerprintMetrics.html">CalculateFingerprintMetrics (Picard)</a>
											</li>
											<li>
												<a href="picard_sam_CalculateReadGroupChecksum.html">CalculateReadGroupChecksum (Picard)</a>
											</li>
											<li>
												<a href="picard_fingerprint_CheckFingerprint.html">CheckFingerprint (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_qc_CheckPileup.html">CheckPileup</a>
											</li>
											<li>
												<a href="picard_sam_CheckTerminatorBlock.html">CheckTerminatorBlock (Picard)</a>
											</li>
											<li>
												<a href="picard_fingerprint_ClusterCrosscheckMetrics.html">ClusterCrosscheckMetrics (Picard)</a>
											</li>
											<li>
												<a href="picard_analysis_CollectAlignmentSummaryMetrics.html">CollectAlignmentSummaryMetrics (Picard)</a>
											</li>
											<li>
												<a href="picard_arrays_CollectArraysVariantCallingMetrics.html">CollectArraysVariantCallingMetrics (Picard)</a>
											</li>
											<li>
												<a href="picard_analysis_CollectBaseDistributionByCycle.html">CollectBaseDistributionByCycle (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_pipelines_metrics_CollectBaseDistributionByCycleSpark.html">CollectBaseDistributionByCycleSpark</a>
											</li>
											<li>
												<a href="picard_analysis_CollectGcBiasMetrics.html">CollectGcBiasMetrics (Picard)</a>
											</li>
											<li>
												<a href="picard_illumina_quality_CollectHiSeqXPfFailMetrics.html">CollectHiSeqXPfFailMetrics (Picard)</a>
											</li>
											<li>
												<a href="picard_analysis_directed_CollectHsMetrics.html">CollectHsMetrics (Picard)</a>
											</li>
											<li>
												<a href="picard_analysis_replicates_CollectIndependentReplicateMetrics.html">CollectIndependentReplicateMetrics (Picard)</a>
											</li>
											<li>
												<a href="picard_analysis_CollectInsertSizeMetrics.html">CollectInsertSizeMetrics (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_pipelines_metrics_CollectInsertSizeMetricsSpark.html">CollectInsertSizeMetricsSpark</a>
											</li>
											<li>
												<a href="picard_analysis_CollectJumpingLibraryMetrics.html">CollectJumpingLibraryMetrics (Picard)</a>
											</li>
											<li>
												<a href="picard_analysis_CollectMultipleMetrics.html">CollectMultipleMetrics (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_pipelines_metrics_CollectMultipleMetricsSpark.html">CollectMultipleMetricsSpark</a>
											</li>
											<li>
												<a href="picard_analysis_CollectOxoGMetrics.html">CollectOxoGMetrics (Picard)</a>
											</li>
											<li>
												<a href="picard_analysis_CollectQualityYieldMetrics.html">CollectQualityYieldMetrics (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_pipelines_metrics_CollectQualityYieldMetricsSpark.html">CollectQualityYieldMetricsSpark</a>
											</li>
											<li>
												<a href="picard_analysis_CollectRawWgsMetrics.html">CollectRawWgsMetrics (Picard)</a>
											</li>
											<li>
												<a href="picard_analysis_CollectRnaSeqMetrics.html">CollectRnaSeqMetrics (Picard)</a>
											</li>
											<li>
												<a href="picard_analysis_CollectRrbsMetrics.html">CollectRrbsMetrics (Picard)</a>
											</li>
											<li>
												<a href="picard_sam_SamErrorMetric_CollectSamErrorMetrics.html">CollectSamErrorMetrics (Picard)</a>
											</li>
											<li>
												<a href="picard_analysis_artifacts_CollectSequencingArtifactMetrics.html">CollectSequencingArtifactMetrics (Picard)</a>
											</li>
											<li>
												<a href="picard_analysis_directed_CollectTargetedPcrMetrics.html">CollectTargetedPcrMetrics (Picard)</a>
											</li>
											<li>
												<a href="picard_vcf_CollectVariantCallingMetrics.html">CollectVariantCallingMetrics (Picard)</a>
											</li>
											<li>
												<a href="picard_analysis_CollectWgsMetrics.html">CollectWgsMetrics (Picard)</a>
											</li>
											<li>
												<a href="picard_analysis_CollectWgsMetricsWithNonZeroCoverage.html">CollectWgsMetricsWithNonZeroCoverage (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_validation_CompareBaseQualities.html">CompareBaseQualities</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_validation_CompareDuplicatesSpark.html">CompareDuplicatesSpark</a>
											</li>
											<li>
												<a href="picard_analysis_CompareMetrics.html">CompareMetrics (Picard)</a>
											</li>
											<li>
												<a href="picard_sam_CompareSAMs.html">CompareSAMs (Picard)</a>
											</li>
											<li>
												<a href="picard_analysis_artifacts_ConvertSequencingArtifactToOxoG.html">ConvertSequencingArtifactToOxoG (Picard)</a>
											</li>
											<li>
												<a href="picard_fingerprint_CrosscheckFingerprints.html">CrosscheckFingerprints (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_DumpTabixIndex.html">DumpTabixIndex</a>
											</li>
											<li>
												<a href="picard_sam_markduplicates_EstimateLibraryComplexity.html">EstimateLibraryComplexity (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_FlagStat.html">FlagStat</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_pipelines_FlagStatSpark.html">FlagStatSpark</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_GetSampleName.html">GetSampleName</a>
											</li>
											<li>
												<a href="picard_analysis_MeanQualityByCycle.html">MeanQualityByCycle (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_pipelines_metrics_MeanQualityByCycleSpark.html">MeanQualityByCycleSpark</a>
											</li>
											<li>
												<a href="picard_analysis_QualityScoreDistribution.html">QualityScoreDistribution (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_pipelines_metrics_QualityScoreDistributionSpark.html">QualityScoreDistributionSpark</a>
											</li>
											<li>
												<a href="picard_sam_ValidateSamFile.html">ValidateSamFile (Picard)</a>
											</li>
											<li>
												<a href="picard_sam_ViewSam.html">ViewSam (Picard)</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#IntervalsManipulation">Intervals Manipulation</a>
							<div id="IntervalsManipulation"
								<?php echo ($group == 'Intervals Manipulation')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_util_BedToIntervalList.html">BedToIntervalList (Picard)</a>
											</li>
											<li>
												<a href="picard_util_IntervalListToBed.html">IntervalListToBed (Picard)</a>
											</li>
											<li>
												<a href="picard_util_IntervalListTools.html">IntervalListTools (Picard)</a>
											</li>
											<li>
												<a href="picard_util_LiftOverIntervalList.html">LiftOverIntervalList (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_copynumber_PreprocessIntervals.html">PreprocessIntervals</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_SplitIntervals.html">SplitIntervals</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#Metagenomics">Metagenomics</a>
							<div id="Metagenomics"
								<?php echo ($group == 'Metagenomics')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_pathseq_PathSeqBuildKmers.html">PathSeqBuildKmers</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_pathseq_PathSeqBuildReferenceTaxonomy.html">PathSeqBuildReferenceTaxonomy</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_pathseq_PathSeqBwaSpark.html">PathSeqBwaSpark</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_pathseq_PathSeqFilterSpark.html">PathSeqFilterSpark</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_pathseq_PathSeqPipelineSpark.html">PathSeqPipelineSpark</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_pathseq_PathSeqScoreSpark.html">PathSeqScoreSpark</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#Other">Other</a>
							<div id="Other"
								<?php echo ($group == 'Other')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_CreateHadoopBamSplittingIndex.html">CreateHadoopBamSplittingIndex</a>
											</li>
											<li>
												<a href="picard_util_FifoBuffer.html">FifoBuffer (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_bqsr_GatherBQSRReports.html">GatherBQSRReports</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_vqsr_GatherTranches.html">GatherTranches</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_IndexFeatureFile.html">IndexFeatureFile</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_ParallelCopyGCSDirectoryIntoHDFSSpark.html">ParallelCopyGCSDirectoryIntoHDFSSpark</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_ReadAnonymizer.html">ReadAnonymizer</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_variantutils_ReblockGVCF.html">ReblockGVCF</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#ReadDataManipulation">Read Data Manipulation</a>
							<div id="ReadDataManipulation"
								<?php echo ($group == 'Read Data Manipulation')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_sam_AddCommentsToBam.html">AddCommentsToBam (Picard)</a>
											</li>
											<li>
												<a href="picard_sam_AddOATag.html">AddOATag (Picard)</a>
											</li>
											<li>
												<a href="picard_sam_AddOrReplaceReadGroups.html">AddOrReplaceReadGroups (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_bqsr_ApplyBQSR.html">ApplyBQSR</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_ApplyBQSRSpark.html">ApplyBQSRSpark</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_pipelines_BQSRPipelineSpark.html">BQSRPipelineSpark</a>
											</li>
											<li>
												<a href="picard_fastq_BamToBfq.html">BamToBfq (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_bqsr_BaseRecalibrator.html">BaseRecalibrator</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_BaseRecalibratorSpark.html">BaseRecalibratorSpark</a>
											</li>
											<li>
												<a href="picard_sam_BuildBamIndex.html">BuildBamIndex (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_pipelines_BwaAndMarkDuplicatesPipelineSpark.html">BwaAndMarkDuplicatesPipelineSpark</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_bwa_BwaSpark.html">BwaSpark</a>
											</li>
											<li>
												<a href="picard_sam_CleanSam.html">CleanSam (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_ClipReads.html">ClipReads</a>
											</li>
											<li>
												<a href="picard_sam_markduplicates_CollectDuplicateMetrics.html">CollectDuplicateMetrics (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_ConvertHeaderlessHadoopBamShardToBam.html">ConvertHeaderlessHadoopBamShardToBam</a>
											</li>
											<li>
												<a href="picard_sam_DownsampleSam.html">DownsampleSam (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_sv_utils_ExtractOriginalAlignmentRecordsByNameSpark.html">ExtractOriginalAlignmentRecordsByNameSpark</a>
											</li>
											<li>
												<a href="picard_sam_FastqToSam.html">FastqToSam (Picard)</a>
											</li>
											<li>
												<a href="picard_sam_FilterSamReads.html">FilterSamReads (Picard)</a>
											</li>
											<li>
												<a href="picard_sam_FixMateInformation.html">FixMateInformation (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_FixMisencodedBaseQualityReads.html">FixMisencodedBaseQualityReads</a>
											</li>
											<li>
												<a href="picard_sam_GatherBamFiles.html">GatherBamFiles (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_LeftAlignIndels.html">LeftAlignIndels</a>
											</li>
											<li>
												<a href="picard_sam_markduplicates_MarkDuplicates.html">MarkDuplicates (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_transforms_markduplicates_MarkDuplicatesSpark.html">MarkDuplicatesSpark</a>
											</li>
											<li>
												<a href="picard_sam_markduplicates_MarkDuplicatesWithMateCigar.html">MarkDuplicatesWithMateCigar (Picard)</a>
											</li>
											<li>
												<a href="picard_sam_MergeBamAlignment.html">MergeBamAlignment (Picard)</a>
											</li>
											<li>
												<a href="picard_sam_MergeSamFiles.html">MergeSamFiles (Picard)</a>
											</li>
											<li>
												<a href="picard_sam_PositionBasedDownsampleSam.html">PositionBasedDownsampleSam (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_qc_PostProcessReadsForRSEM.html">PostProcessReadsForRSEM</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_PrintDistantMates.html">PrintDistantMates</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_PrintReads.html">PrintReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_PrintReadsHeader.html">PrintReadsHeader</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_pipelines_PrintReadsSpark.html">PrintReadsSpark</a>
											</li>
											<li>
												<a href="picard_sam_ReorderSam.html">ReorderSam (Picard)</a>
											</li>
											<li>
												<a href="picard_sam_ReplaceSamHeader.html">ReplaceSamHeader (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_RevertBaseQualityScores.html">RevertBaseQualityScores</a>
											</li>
											<li>
												<a href="picard_sam_RevertOriginalBaseQualitiesAndAddMateCigar.html">RevertOriginalBaseQualitiesAndAddMateCigar (Picard)</a>
											</li>
											<li>
												<a href="picard_sam_RevertSam.html">RevertSam (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_RevertSamSpark.html">RevertSamSpark</a>
											</li>
											<li>
												<a href="picard_sam_SamFormatConverter.html">SamFormatConverter (Picard)</a>
											</li>
											<li>
												<a href="picard_sam_SamToFastq.html">SamToFastq (Picard)</a>
											</li>
											<li>
												<a href="picard_sam_SetNmAndUqTags.html">SetNmAndUqTags (Picard)</a>
											</li>
											<li>
												<a href="picard_sam_SetNmMdAndUqTags.html">SetNmMdAndUqTags (Picard)</a>
											</li>
											<li>
												<a href="picard_sam_markduplicates_SimpleMarkDuplicatesWithMateCigar.html">SimpleMarkDuplicatesWithMateCigar (Picard)</a>
											</li>
											<li>
												<a href="picard_sam_SortSam.html">SortSam (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_pipelines_SortSamSpark.html">SortSamSpark</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_rnaseq_SplitNCigarReads.html">SplitNCigarReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_SplitReads.html">SplitReads</a>
											</li>
											<li>
												<a href="picard_sam_SplitSamByLibrary.html">SplitSamByLibrary (Picard)</a>
											</li>
											<li>
												<a href="picard_sam_SplitSamByNumberOfReads.html">SplitSamByNumberOfReads (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_qc_TransferReadTags.html">TransferReadTags</a>
											</li>
											<li>
												<a href="picard_sam_markduplicates_UmiAwareMarkDuplicatesWithMateCigar.html">UmiAwareMarkDuplicatesWithMateCigar (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_UnmarkDuplicates.html">UnmarkDuplicates</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#Reference">Reference</a>
							<div id="Reference"
								<?php echo ($group == 'Reference')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_util_BaitDesigner.html">BaitDesigner (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_BwaMemIndexImageCreator.html">BwaMemIndexImageCreator</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_reference_CheckReferenceCompatibility.html">CheckReferenceCompatibility</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_reference_CompareReferences.html">CompareReferences</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_dragstr_ComposeSTRTableFile.html">ComposeSTRTableFile</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_fasta_CountBasesInReference.html">CountBasesInReference</a>
											</li>
											<li>
												<a href="picard_sam_CreateSequenceDictionary.html">CreateSequenceDictionary (Picard)</a>
											</li>
											<li>
												<a href="picard_reference_ExtractSequences.html">ExtractSequences (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_fasta_FastaAlternateReferenceMaker.html">FastaAlternateReferenceMaker</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_fasta_FastaReferenceMaker.html">FastaReferenceMaker</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_sv_evidence_FindBadGenomicKmersSpark.html">FindBadGenomicKmersSpark</a>
											</li>
											<li>
												<a href="picard_reference_NonNFastaSize.html">NonNFastaSize (Picard)</a>
											</li>
											<li>
												<a href="picard_reference_NormalizeFasta.html">NormalizeFasta (Picard)</a>
											</li>
											<li>
												<a href="picard_util_ScatterIntervalsByNs.html">ScatterIntervalsByNs (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_fasta_ShiftFasta.html">ShiftFasta</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#ShortVariantDiscovery">Short Variant Discovery</a>
							<div id="ShortVariantDiscovery"
								<?php echo ($group == 'Short Variant Discovery')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="org_broadinstitute_hellbender_tools_dragstr_CalibrateDragstrModel.html">CalibrateDragstrModel</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_CombineGVCFs.html">CombineGVCFs</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_genomicsdb_GenomicsDBImport.html">GenomicsDBImport</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_GenotypeGVCFs.html">GenotypeGVCFs</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_gnarlyGenotyper_GnarlyGenotyper.html">GnarlyGenotyper</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_variantrecalling_HaplotypeBasedVariantRecaller.html">HaplotypeBasedVariantRecaller</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_haplotypecaller_HaplotypeCaller.html">HaplotypeCaller</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_HaplotypeCallerSpark.html">HaplotypeCallerSpark</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_readorientation_LearnReadOrientationModel.html">LearnReadOrientationModel</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_mutect_Mutect2.html">Mutect2</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_haplotypecaller_RampedHaplotypeCaller.html">RampedHaplotypeCaller</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_pipelines_ReadsPipelineSpark.html">ReadsPipelineSpark</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#StructuralVariantDiscovery">Structural Variant Discovery</a>
							<div id="StructuralVariantDiscovery"
								<?php echo ($group == 'Structural Variant Discovery')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_sv_CollectSVEvidence.html">CollectSVEvidence</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_sv_discovery_inference_CpxVariantReInterpreterSpark.html">CpxVariantReInterpreterSpark</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_sv_discovery_DiscoverVariantsFromContigAlignmentsSAMSpark.html">DiscoverVariantsFromContigAlignmentsSAMSpark</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_sv_evidence_ExtractSVEvidenceSpark.html">ExtractSVEvidenceSpark</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_sv_evidence_FindBreakpointEvidenceSpark.html">FindBreakpointEvidenceSpark</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_sv_JointGermlineCNVSegmentation.html">JointGermlineCNVSegmentation</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_sv_PrintReadCounts.html">PrintReadCounts</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_sv_SVAnnotate.html">SVAnnotate</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_sv_SVCluster.html">SVCluster</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_sv_SiteDepthtoBAF.html">SiteDepthtoBAF</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_sv_StructuralVariationDiscoveryPipelineSpark.html">StructuralVariationDiscoveryPipelineSpark</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_sv_discovery_SvDiscoverFromLocalAssemblyContigAlignmentsSpark.html">SvDiscoverFromLocalAssemblyContigAlignmentsSpark</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantEvaluationandRefinement">Variant Evaluation and Refinement</a>
							<div id="VariantEvaluationandRefinement"
								<?php echo ($group == 'Variant Evaluation and Refinement')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_validation_AnnotateVcfWithBamDepth.html">AnnotateVcfWithBamDepth</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_validation_AnnotateVcfWithExpectedAlleleFraction.html">AnnotateVcfWithExpectedAlleleFraction</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_variantutils_CalculateGenotypePosteriors.html">CalculateGenotypePosteriors</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_validation_CalculateMixingFractions.html">CalculateMixingFractions</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_validation_Concordance.html">Concordance</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_validation_CountFalsePositives.html">CountFalsePositives</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_CountVariants.html">CountVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_pipelines_CountVariantsSpark.html">CountVariantsSpark</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_validation_EvaluateInfoFieldConcordance.html">EvaluateInfoFieldConcordance</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_funcotator_FilterFuncotations.html">FilterFuncotations</a>
											</li>
											<li>
												<a href="picard_vcf_MendelianViolations_FindMendelianViolations.html">FindMendelianViolations (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_funcotator_FuncotateSegments.html">FuncotateSegments</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_funcotator_Funcotator.html">Funcotator</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_funcotator_FuncotatorDataSourceDownloader.html">FuncotatorDataSourceDownloader</a>
											</li>
											<li>
												<a href="picard_vcf_GenotypeConcordance.html">GenotypeConcordance (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_validation_basicshortmutpileup_ValidateBasicSomaticShortMutations.html">ValidateBasicSomaticShortMutations</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_variantutils_ValidateVariants.html">ValidateVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_varianteval_VariantEval.html">VariantEval</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_variantutils_VariantsToTable.html">VariantsToTable</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantFiltering">Variant Filtering</a>
							<div id="VariantFiltering"
								<?php echo ($group == 'Variant Filtering')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_vqsr_ApplyVQSR.html">ApplyVQSR</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_vqsr_CNNScoreVariants.html">CNNScoreVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_vqsr_CNNVariantTrain.html">CNNVariantTrain</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_vqsr_CNNVariantWriteTensors.html">CNNVariantWriteTensors</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_mutect_CreateSomaticPanelOfNormals.html">CreateSomaticPanelOfNormals</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_vqsr_scalable_ExtractVariantAnnotations.html">ExtractVariantAnnotations</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_realignmentfilter_FilterAlignmentArtifacts.html">FilterAlignmentArtifacts</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_mutect_filtering_FilterMutectCalls.html">FilterMutectCalls</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_vqsr_FilterVariantTranches.html">FilterVariantTranches</a>
											</li>
											<li>
												<a href="picard_vcf_filter_FilterVcf.html">FilterVcf (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_vqsr_scalable_ScoreVariantAnnotations.html">ScoreVariantAnnotations</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_vqsr_scalable_TrainVariantAnnotationsModel.html">TrainVariantAnnotationsModel</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_filters_VariantFiltration.html">VariantFiltration</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_vqsr_VariantRecalibrator.html">VariantRecalibrator</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantManipulation">Variant Manipulation</a>
							<div id="VariantManipulation"
								<?php echo ($group == 'Variant Manipulation')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_vcf_FixVcfHeader.html">FixVcfHeader (Picard)</a>
											</li>
											<li>
												<a href="picard_vcf_GatherVcfs.html">GatherVcfs (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_GatherVcfsCloud.html">GatherVcfsCloud</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_variantutils_LeftAlignAndTrimVariants.html">LeftAlignAndTrimVariants</a>
											</li>
											<li>
												<a href="picard_vcf_LiftoverVcf.html">LiftoverVcf (Picard)</a>
											</li>
											<li>
												<a href="picard_vcf_MakeSitesOnlyVcf.html">MakeSitesOnlyVcf (Picard)</a>
											</li>
											<li>
												<a href="picard_vcf_MergeVcfs.html">MergeVcfs (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_pipelines_PrintVariantsSpark.html">PrintVariantsSpark</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_validation_RemoveNearbyIndels.html">RemoveNearbyIndels</a>
											</li>
											<li>
												<a href="picard_vcf_RenameSampleInVcf.html">RenameSampleInVcf (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_variantutils_SelectVariants.html">SelectVariants</a>
											</li>
											<li>
												<a href="picard_vcf_SortVcf.html">SortVcf (Picard)</a>
											</li>
											<li>
												<a href="picard_vcf_SplitVcfs.html">SplitVcfs (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_variantutils_UpdateVCFSequenceDictionary.html">UpdateVCFSequenceDictionary</a>
											</li>
											<li>
												<a href="picard_vcf_UpdateVcfSequenceDictionary.html">UpdateVcfSequenceDictionary (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_VariantAnnotator.html">VariantAnnotator</a>
											</li>
											<li>
												<a href="picard_vcf_VcfFormatConverter.html">VcfFormatConverter (Picard)</a>
											</li>
											<li>
												<a href="picard_vcf_VcfToIntervalList.html">VcfToIntervalList (Picard)</a>
											</li>
								</ul>
							</div>
						</li>
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#BaseCalling">Base Calling</a>
							<div id="BaseCalling"
								<?php echo ($group == 'Base Calling')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_illumina_CheckIlluminaDirectory.html">CheckIlluminaDirectory (Picard)</a>
											</li>
											<li>
												<a href="picard_illumina_CollectIlluminaBasecallingMetrics.html">CollectIlluminaBasecallingMetrics (Picard)</a>
											</li>
											<li>
												<a href="picard_illumina_CollectIlluminaLaneMetrics.html">CollectIlluminaLaneMetrics (Picard)</a>
											</li>
											<li>
												<a href="picard_illumina_ExtractIlluminaBarcodes.html">ExtractIlluminaBarcodes (Picard)</a>
											</li>
											<li>
												<a href="picard_illumina_IlluminaBasecallsToFastq.html">IlluminaBasecallsToFastq (Picard)</a>
											</li>
											<li>
												<a href="picard_illumina_IlluminaBasecallsToSam.html">IlluminaBasecallsToSam (Picard)</a>
											</li>
											<li>
												<a href="picard_illumina_MarkIlluminaAdapters.html">MarkIlluminaAdapters (Picard)</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#FlowBasedTools">Flow Based Tools</a>
							<div id="FlowBasedTools"
								<?php echo ($group == 'Flow Based Tools')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_featuremapping_FlowFeatureMapper.html">FlowFeatureMapper</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_groundtruth_GroundTruthReadsBuilder.html">GroundTruthReadsBuilder</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#GenotypingArraysManipulation">Genotyping Arrays Manipulation</a>
							<div id="GenotypingArraysManipulation"
								<?php echo ($group == 'Genotyping Arrays Manipulation')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_arrays_illumina_BpmToNormalizationManifestCsv.html">BpmToNormalizationManifestCsv (Picard)</a>
											</li>
											<li>
												<a href="picard_arrays_CombineGenotypingArrayVcfs.html">CombineGenotypingArrayVcfs (Picard)</a>
											</li>
											<li>
												<a href="picard_arrays_CreateBafRegressMetricsFile.html">CreateBafRegressMetricsFile (Picard)</a>
											</li>
											<li>
												<a href="picard_arrays_CreateVerifyIDIntensityContaminationMetricsFile.html">CreateVerifyIDIntensityContaminationMetricsFile (Picard)</a>
											</li>
											<li>
												<a href="picard_arrays_GtcToVcf.html">GtcToVcf (Picard)</a>
											</li>
											<li>
												<a href="picard_arrays_MergePedIntoVcf.html">MergePedIntoVcf (Picard)</a>
											</li>
											<li>
												<a href="picard_arrays_VcfToAdpc.html">VcfToAdpc (Picard)</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#MethylationSpecificTools">Methylation-Specific Tools</a>
							<div id="MethylationSpecificTools"
								<?php echo ($group == 'Methylation-Specific Tools')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_MethylationTypeCaller.html">MethylationTypeCaller</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#Metrics">Metrics</a>
							<div id="Metrics"
								<?php echo ($group == 'Metrics')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_analysis_AlignmentSummaryMetrics.html">AlignmentSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_arrays_CollectArraysVariantCallingMetrics$ArraysControlCodesSummaryMetrics.html">ArraysControlCodesSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_arrays_CollectArraysVariantCallingMetrics$ArraysVariantCallingDetailMetrics.html">ArraysVariantCallingDetailMetrics</a>
											</li>
											<li>
												<a href="picard_arrays_CollectArraysVariantCallingMetrics$ArraysVariantCallingSummaryMetrics.html">ArraysVariantCallingSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_arrays_BafRegressMetrics.html">BafRegressMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_artifacts_SequencingArtifactMetrics$BaitBiasDetailMetrics.html">BaitBiasDetailMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_artifacts_SequencingArtifactMetrics$BaitBiasSummaryMetrics.html">BaitBiasSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_illumina_BarcodeMetric.html">BarcodeMetric</a>
											</li>
											<li>
												<a href="picard_analysis_BaseDistributionByCycleMetrics.html">BaseDistributionByCycleMetrics</a>
											</li>
											<li>
												<a href="picard_sam_SamErrorMetric_BaseErrorMetric.html">BaseErrorMetric</a>
											</li>
											<li>
												<a href="picard_fingerprint_ClusteredCrosscheckMetric.html">ClusteredCrosscheckMetric</a>
											</li>
											<li>
												<a href="picard_analysis_CollectOxoGMetrics$CpcgMetrics.html">CpcgMetrics</a>
											</li>
											<li>
												<a href="picard_sam_DuplicationMetrics.html">DuplicationMetrics</a>
											</li>
											<li>
												<a href="picard_sam_SamErrorMetric_ErrorMetric.html">ErrorMetric</a>
											</li>
											<li>
												<a href="picard_analysis_artifacts_ErrorSummaryMetrics.html">ErrorSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_fingerprint_FingerprintMetrics.html">FingerprintMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_FingerprintingDetailMetrics.html">FingerprintingDetailMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_FingerprintingSummaryMetrics.html">FingerprintingSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_GcBiasDetailMetrics.html">GcBiasDetailMetrics</a>
											</li>
											<li>
												<a href="picard_metrics_GcBiasMetrics.html">GcBiasMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_GcBiasSummaryMetrics.html">GcBiasSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_vcf_GenotypeConcordanceContingencyMetrics.html">GenotypeConcordanceContingencyMetrics</a>
											</li>
											<li>
												<a href="picard_vcf_GenotypeConcordanceDetailMetrics.html">GenotypeConcordanceDetailMetrics</a>
											</li>
											<li>
												<a href="picard_vcf_GenotypeConcordanceSummaryMetrics.html">GenotypeConcordanceSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_directed_HsMetrics.html">HsMetrics</a>
											</li>
											<li>
												<a href="picard_illumina_IlluminaBasecallingMetrics.html">IlluminaBasecallingMetrics</a>
											</li>
											<li>
												<a href="picard_illumina_IlluminaLaneMetrics.html">IlluminaLaneMetrics</a>
											</li>
											<li>
												<a href="picard_illumina_IlluminaPhasingMetrics.html">IlluminaPhasingMetrics</a>
											</li>
											<li>
												<a href="picard_sam_SamErrorMetric_IndelErrorMetric.html">IndelErrorMetric</a>
											</li>
											<li>
												<a href="picard_analysis_replicates_IndependentReplicateMetric.html">IndependentReplicateMetric</a>
											</li>
											<li>
												<a href="picard_analysis_InsertSizeMetrics.html">InsertSizeMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_JumpingLibraryMetrics.html">JumpingLibraryMetrics</a>
											</li>
											<li>
												<a href="picard_vcf_MendelianViolations_MendelianViolationMetrics.html">MendelianViolationMetrics</a>
											</li>
											<li>
												<a href="picard_sam_SamErrorMetric_OverlappingErrorMetric.html">OverlappingErrorMetric</a>
											</li>
											<li>
												<a href="picard_illumina_quality_CollectHiSeqXPfFailMetrics$PFFailDetailedMetric.html">PFFailDetailedMetric</a>
											</li>
											<li>
												<a href="picard_illumina_quality_CollectHiSeqXPfFailMetrics$PFFailSummaryMetric.html">PFFailSummaryMetric</a>
											</li>
											<li>
												<a href="picard_analysis_artifacts_SequencingArtifactMetrics$PreAdapterDetailMetrics.html">PreAdapterDetailMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_artifacts_SequencingArtifactMetrics$PreAdapterSummaryMetrics.html">PreAdapterSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectQualityYieldMetrics$QualityYieldMetrics.html">QualityYieldMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectRawWgsMetrics$RawWgsMetrics.html">RawWgsMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_RnaSeqMetrics.html">RnaSeqMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_RrbsCpgDetailMetrics.html">RrbsCpgDetailMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_RrbsSummaryMetrics.html">RrbsSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_sam_SamComparisonMetric.html">SamComparisonMetric</a>
											</li>
											<li>
												<a href="picard_analysis_directed_TargetMetrics.html">TargetMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_directed_TargetedPcrMetrics.html">TargetedPcrMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_TheoreticalSensitivityMetrics.html">TheoreticalSensitivityMetrics</a>
											</li>
											<li>
												<a href="picard_sam_markduplicates_UmiMetrics.html">UmiMetrics</a>
											</li>
											<li>
												<a href="picard_vcf_CollectVariantCallingMetrics$VariantCallingDetailMetrics.html">VariantCallingDetailMetrics</a>
											</li>
											<li>
												<a href="picard_vcf_CollectVariantCallingMetrics$VariantCallingSummaryMetrics.html">VariantCallingSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_arrays_VerifyIDIntensityContaminationMetrics.html">VerifyIDIntensityContaminationMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_WgsMetrics.html">WgsMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectWgsMetricsWithNonZeroCoverage$WgsMetricsWithNonZeroCoverage.html">WgsMetricsWithNonZeroCoverage</a>
											</li>
								</ul>
							</div>
						</li>
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#FlowAnnotations">Flow Annotations</a>
							<div id="FlowAnnotations"
								<?php echo ($group == 'Flow Annotations')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_flow_CycleSkipStatus.html">CycleSkipStatus</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_flow_GcContent.html">GcContent</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_HaplotypeFilteringAnnotation.html">HaplotypeFilteringAnnotation</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_flow_HmerIndelLength.html">HmerIndelLength</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_flow_HmerIndelNuc.html">HmerIndelNuc</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_flow_HmerMotifs.html">HmerMotifs</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_flow_IndelClassify.html">IndelClassify</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_flow_IndelLength.html">IndelLength</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_flow_VariantType.html">VariantType</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#ReadFilters">Read Filters</a>
							<div id="ReadFilters"
								<?php echo ($group == 'Read Filters')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_AlignmentAgreesWithHeaderReadFilter.html">AlignmentAgreesWithHeaderReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$AllowAllReadsReadFilter.html">AllowAllReadsReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_AmbiguousBaseReadFilter.html">AmbiguousBaseReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$CigarContainsNoNOperator.html">CigarContainsNoNOperator</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ExcessiveEndClippedReadFilter.html">ExcessiveEndClippedReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$FirstOfPairReadFilter.html">FirstOfPairReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_flow_FlowBasedTPAttributeSymetricReadFilter.html">FlowBasedTPAttributeSymetricReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_flow_FlowBasedTPAttributeValidReadFilter.html">FlowBasedTPAttributeValidReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_FragmentLengthReadFilter.html">FragmentLengthReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$GoodCigarReadFilter.html">GoodCigarReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$HasReadGroupReadFilter.html">HasReadGroupReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_flow_HmerQualitySymetricReadFilter.html">HmerQualitySymetricReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_IntervalOverlapReadFilter.html">IntervalOverlapReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_JexlExpressionReadTagValueFilter.html">JexlExpressionReadTagValueFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_LibraryReadFilter.html">LibraryReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$MappedReadFilter.html">MappedReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$MappingQualityAvailableReadFilter.html">MappingQualityAvailableReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$MappingQualityNotZeroReadFilter.html">MappingQualityNotZeroReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_MappingQualityReadFilter.html">MappingQualityReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$MatchingBasesAndQualsReadFilter.html">MatchingBasesAndQualsReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$MateDifferentStrandReadFilter.html">MateDifferentStrandReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_MateDistantReadFilter.html">MateDistantReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$MateOnSameContigOrNoMappedMateReadFilter.html">MateOnSameContigOrNoMappedMateReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$MateUnmappedAndUnmappedReadFilter.html">MateUnmappedAndUnmappedReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_MetricsReadFilter.html">MetricsReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$NonChimericOriginalAlignmentReadFilter.html">NonChimericOriginalAlignmentReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$NonZeroFragmentLengthReadFilter.html">NonZeroFragmentLengthReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$NonZeroReferenceLengthAlignmentReadFilter.html">NonZeroReferenceLengthAlignmentReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$NotDuplicateReadFilter.html">NotDuplicateReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$NotProperlyPairedReadFilter.html">NotProperlyPairedReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$NotSecondaryAlignmentReadFilter.html">NotSecondaryAlignmentReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$NotSupplementaryAlignmentReadFilter.html">NotSupplementaryAlignmentReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_OverclippedReadFilter.html">OverclippedReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$PairedReadFilter.html">PairedReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$PassesVendorQualityCheckReadFilter.html">PassesVendorQualityCheckReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_PlatformReadFilter.html">PlatformReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_PlatformUnitReadFilter.html">PlatformUnitReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$PrimaryLineReadFilter.html">PrimaryLineReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$ProperlyPairedReadFilter.html">ProperlyPairedReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadGroupBlackListReadFilter.html">ReadGroupBlackListReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_flow_ReadGroupHasFlowOrderReadFilter.html">ReadGroupHasFlowOrderReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadGroupReadFilter.html">ReadGroupReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$ReadLengthEqualsCigarLengthReadFilter.html">ReadLengthEqualsCigarLengthReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadLengthReadFilter.html">ReadLengthReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadNameReadFilter.html">ReadNameReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadStrandFilter.html">ReadStrandFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadTagValueFilter.html">ReadTagValueFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_SampleReadFilter.html">SampleReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$SecondOfPairReadFilter.html">SecondOfPairReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$SeqIsStoredReadFilter.html">SeqIsStoredReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_SoftClippedReadFilter.html">SoftClippedReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$ValidAlignmentEndReadFilter.html">ValidAlignmentEndReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$ValidAlignmentStartReadFilter.html">ValidAlignmentStartReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_flow_WellformedFlowBasedReadFilter.html">WellformedFlowBasedReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_WellformedReadFilter.html">WellformedReadFilter</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantAnnotations">Variant Annotations</a>
							<div id="VariantAnnotations"
								<?php echo ($group == 'Variant Annotations')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_allelespecific_AS_BaseQualityRankSumTest.html">AS_BaseQualityRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_allelespecific_AS_FisherStrand.html">AS_FisherStrand</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_allelespecific_AS_InbreedingCoeff.html">AS_InbreedingCoeff</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_allelespecific_AS_MappingQualityRankSumTest.html">AS_MappingQualityRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_allelespecific_AS_QualByDepth.html">AS_QualByDepth</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_allelespecific_AS_RMSMappingQuality.html">AS_RMSMappingQuality</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_allelespecific_AS_ReadPosRankSumTest.html">AS_ReadPosRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_allelespecific_AS_StrandOddsRatio.html">AS_StrandOddsRatio</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_AlleleFraction.html">AlleleFraction</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_AllelePseudoDepth.html">AllelePseudoDepth</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_AssemblyComplexity.html">AssemblyComplexity</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_BaseQuality.html">BaseQuality</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_BaseQualityRankSumTest.html">BaseQualityRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_ChromosomeCounts.html">ChromosomeCounts</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_ClippingRankSumTest.html">ClippingRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_CountNs.html">CountNs</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_Coverage.html">Coverage</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_DepthPerAlleleBySample.html">DepthPerAlleleBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_DepthPerSampleHC.html">DepthPerSampleHC</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_ExcessHet.html">ExcessHet</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_FeaturizedReadSets.html">FeaturizedReadSets</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_FisherStrand.html">FisherStrand</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_FragmentDepthPerAlleleBySample.html">FragmentDepthPerAlleleBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_FragmentLength.html">FragmentLength</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_GenotypeSummaries.html">GenotypeSummaries</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_InbreedingCoeff.html">InbreedingCoeff</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_LikelihoodRankSumTest.html">LikelihoodRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_MappingQuality.html">MappingQuality</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_MappingQualityRankSumTest.html">MappingQualityRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_MappingQualityZero.html">MappingQualityZero</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_OrientationBiasReadCounts.html">OrientationBiasReadCounts</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_OriginalAlignment.html">OriginalAlignment</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_PossibleDeNovo.html">PossibleDeNovo</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_QualByDepth.html">QualByDepth</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_RMSMappingQuality.html">RMSMappingQuality</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_RawGtCount.html">RawGtCount</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_ReadPosRankSumTest.html">ReadPosRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_ReadPosition.html">ReadPosition</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_ReferenceBases.html">ReferenceBases</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_SampleList.html">SampleList</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_StrandBiasBySample.html">StrandBiasBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_StrandOddsRatio.html">StrandOddsRatio</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_TandemRepeat.html">TandemRepeat</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_UniqueAltReadCount.html">UniqueAltReadCount</a>
											</li>
								</ul>
							</div>
						</li>
        </ul>
		</aside>

	</section>

	<div class="col-md-8">

			<h1>**EXPERIMENTAL** GroundTruthReadsBuilder</h1>

			<p>Produces a flexible and robust ground truth set for base calling training</p>

			<h3>Category
				<small> Flow Based Tools</small>
			</h3>
			<h3>Traversal
				<small>ReadWalker</small>
			</h3>
		<hr>
		<h2>Overview</h2>
		An internal tool to produce a flexible and robust ground truth set for base calling training.


 <h3> Input </h3>
 <ul>
     <li> Coordinate-sorted and indexed SAM/BAM/CRAM </li>
     <li> Maternal and Parental references (fa) </li>
     <li> Folder with address translation files from reference to maternal/parental references (filename example: maternal.chr9.csv)</li>
 </ul>

 <h3> Output </h3>
 <ul>
     <li> CSV file containing maternal/parental haplotype scores and many more columns (.csv or .csv.gz supported) </li>
 </ul>

 At present, the output will contain the following columns (not nessarily in this order) for each processed read:
 <ul>
     <li>ReadName, ReadChrom, ReadStart, ReadEnd, tm, mapq, flags, ReadCigar, ReadSequence, ReadUnclippedStart,
     ReadUnclippedEnd - information directly extracted from the input read</li>
     <li>PaternalHaplotypeInterval, BestHaplotypeSequence, PaternalHaplotypeScore - parental haplotype information.
     First the read interval is translated into parental space, the haplotype sequence extracted and then scored</li>
     <li>Maternal* - same for maternal</li>
     <li>RefHaplotypeScore - score computed from the reference haplotype</li>
     <li>BestHaplotypeKey - flow based key for the 'best' (score wise) haplotype (out of the maternal/paternal) pair)</li>
     <li>ConsensusHaplotypeKey - a flow based key constructed from the flow keys of the maternal and paternal
     haplotypes, containing only keys that agree (other keys filled with fixed/special value)</li>
 </ul>


 <h3>Usage examples</h3>
 <pre>
 gatk GroundTruthReadsBuilder \
  -R
  ../../../ref/Homo_sapiens_assembly38.fasta
  -I
  150548-UGAv3-4.chr9.cram
  --maternal-ref
  chr9_HG001_maternal.fa
  --paternal-ref
  chr9_HG001_paternal.fa
  --ancestral-translators-base-path
  ./
  --output-csv
  output-small.csv
  --subsampling-ratio
  1.0
  --max-output-reads
  100000000
  --intervals
  chr9:109991494-109991494
  --smith-waterman
  FASTEST_AVAILABLE
  --likelihood-calculation-engine
  FlowBased
  -mbq
  0
  --kmer-size
  10
  --gt-debug
  --output-flow-length
  1000
  --haplotype-output-padding-size
  8
  --prepend-sequence
  TTTT
  --append-sequence
  CCCC
 </pre>

 

				<hr>
				<h2>Additional Information</h2>
				<p></p>
				<h3>Read filters</h3>
					<p>This Read Filter is automatically applied to the data by the Engine before processing by GroundTruthReadsBuilder.</p>
				<ul>
						<li><a href="org_broadinstitute_hellbender_engine_filters_WellformedReadFilter.html">WellformedReadFilter</a></li>
				</ul>

				<h3>GroundTruthReadsBuilder specific arguments</h3>
				<p>This table summarizes the command-line arguments that are specific to this tool. For more details on each argument, see the list further down below the table or click on an argument name to jump directly to that entry in the list.</p>
				<table class="table table-striped table-bordered table-condensed cozy">
					<thead>
					<tr>
						<th>Argument name(s)</th>
						<th>Default value</th>
						<th>Summary</th>
					</tr>
					</thead>
					<tbody>
			<tr>
				<th colspan="4" id="row-divider">Required Arguments</th>
			</tr>
				<tr>
					<td><a href="#--ancestral-translators-base-path">--ancestral-translators-base-path</a><br />
					</td>
					<!--<td>GATKPath</td> -->
						<td></td>
					<td>base path for ancestral translation ancestral.contig.csv files</td>
				</tr>
				<tr>
					<td><a href="#--input">--input</a><br />
								&nbsp;<em>-I</em>
					</td>
					<!--<td>List[GATKPath]</td> -->
						<td></td>
					<td>BAM/SAM/CRAM file containing reads</td>
				</tr>
				<tr>
					<td><a href="#--maternal-ref">--maternal-ref</a><br />
					</td>
					<!--<td>GATKPath</td> -->
						<td></td>
					<td>maternal reference file</td>
				</tr>
				<tr>
					<td><a href="#--output-csv">--output-csv</a><br />
					</td>
					<!--<td>GATKPath</td> -->
						<td></td>
					<td>main CSV output file. the file containing maternal/parental maternal and paternal haplotype sequences and scores (and many more columns). supported file extensions: .csv, .csv.gz.</td>
				</tr>
				<tr>
					<td><a href="#--paternal-ref">--paternal-ref</a><br />
					</td>
					<!--<td>GATKPath</td> -->
						<td></td>
					<td>paternal reference file</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Optional Tool Arguments</th>
			</tr>
				<tr>
					<td><a href="#--append-sequence">--append-sequence</a><br />
					</td>
					<!--<td>String</td> -->
						<td></td>
					<td>Sequence to append (adapter)</td>
				</tr>
				<tr>
					<td><a href="#--arguments_file">--arguments_file</a><br />
					</td>
					<!--<td>List[File]</td> -->
						<td></td>
					<td>read one or more arguments files and add them to the command line</td>
				</tr>
				<tr>
					<td><a href="#--base-quality-score-threshold">--base-quality-score-threshold</a><br />
					</td>
					<!--<td>byte</td> -->
						<td>18</td>
					<td>Base qualities below this threshold will be reduced to the minimum (6)</td>
				</tr>
				<tr>
					<td><a href="#--cloud-index-prefetch-buffer">--cloud-index-prefetch-buffer</a><br />
								&nbsp;<em>-CIPB</em>
					</td>
					<!--<td>int</td> -->
						<td>-1</td>
					<td>Size of the cloud-only prefetch buffer (in MB; 0 to disable). Defaults to cloudPrefetchBuffer if unset.</td>
				</tr>
				<tr>
					<td><a href="#--cloud-prefetch-buffer">--cloud-prefetch-buffer</a><br />
								&nbsp;<em>-CPB</em>
					</td>
					<!--<td>int</td> -->
						<td>40</td>
					<td>Size of the cloud-only prefetch buffer (in MB; 0 to disable).</td>
				</tr>
				<tr>
					<td><a href="#--disable-bam-index-caching">--disable-bam-index-caching</a><br />
								&nbsp;<em>-DBIC</em>
					</td>
					<!--<td>boolean</td> -->
						<td>false</td>
					<td>If true, don't cache bam indexes, this will reduce memory requirements but may harm performance if many intervals are specified.  Caching is automatically disabled if there are no intervals specified.</td>
				</tr>
				<tr>
					<td><a href="#--disable-sequence-dictionary-validation">--disable-sequence-dictionary-validation</a><br />
					</td>
					<!--<td>boolean</td> -->
						<td>false</td>
					<td>If specified, do not check the sequence dictionaries from our inputs for compatibility. Use at your own risk!</td>
				</tr>
				<tr>
					<td><a href="#--discard-non-polyt-softclipped-reads">--discard-non-polyt-softclipped-reads</a><br />
					</td>
					<!--<td>boolean</td> -->
						<td>false</td>
					<td>Discard reads which are softclipped, unless the softclip is polyT, defaults to true</td>
				</tr>
				<tr>
					<td><a href="#--dont-use-dragstr-pair-hmm-scores">--dont-use-dragstr-pair-hmm-scores</a><br />
					</td>
					<!--<td>boolean</td> -->
						<td>false</td>
					<td>disable DRAGstr pair-hmm score even when dragstr-params-path was provided</td>
				</tr>
				<tr>
					<td><a href="#--dragstr-het-hom-ratio">--dragstr-het-hom-ratio</a><br />
					</td>
					<!--<td>int</td> -->
						<td>2</td>
					<td>het to hom prior ratio use with DRAGstr on</td>
				</tr>
				<tr>
					<td><a href="#--dragstr-params-path">--dragstr-params-path</a><br />
					</td>
					<!--<td>GATKPath</td> -->
						<td></td>
					<td>location of the DRAGstr model parameters for STR error correction used in the Pair HMM. When provided, it overrides other PCR error correcting mechanisms</td>
				</tr>
				<tr>
					<td><a href="#--enable-dynamic-read-disqualification-for-genotyping">--enable-dynamic-read-disqualification-for-genotyping</a><br />
					</td>
					<!--<td>boolean</td> -->
						<td>false</td>
					<td>Will enable less strict read disqualification low base quality reads</td>
				</tr>
				<tr>
					<td><a href="#--false-snp-compensation">--false-snp-compensation</a><br />
					</td>
					<!--<td>boolean</td> -->
						<td>false</td>
					<td>skip haplotype bases until same base as read starts (false SNP compensation)</td>
				</tr>
				<tr>
					<td><a href="#--fill-softclipped-reads">--fill-softclipped-reads</a><br />
					</td>
					<!--<td>boolean</td> -->
						<td>false</td>
					<td>Softclipped reads should be filled from haplotype, otherwise (default) filled with -83</td>
				</tr>
				<tr>
					<td><a href="#--fill-trimmed-reads">--fill-trimmed-reads</a><br />
					</td>
					<!--<td>boolean</td> -->
						<td>false</td>
					<td>Reads with tm:Q or tm:Z should be filled from haplotype, otherwise (default) filled with -80</td>
				</tr>
				<tr>
					<td><a href="#--fill-trimmed-reads-Q">--fill-trimmed-reads-Q</a><br />
					</td>
					<!--<td>boolean</td> -->
						<td>false</td>
					<td>Reads with tm:Q should be filled from haplotype, otherwise (default) filled with -80</td>
				</tr>
				<tr>
					<td><a href="#--fill-trimmed-reads-Z">--fill-trimmed-reads-Z</a><br />
					</td>
					<!--<td>boolean</td> -->
						<td>false</td>
					<td>Reads with tm:Z should be filled from haplotype, otherwise (default) filled with -80</td>
				</tr>
				<tr>
					<td><a href="#--gcs-max-retries">--gcs-max-retries</a><br />
								&nbsp;<em>-gcs-retries</em>
					</td>
					<!--<td>int</td> -->
						<td>20</td>
					<td>If the GCS bucket channel errors out, how many times it will attempt to re-initiate the connection</td>
				</tr>
				<tr>
					<td><a href="#--gcs-project-for-requester-pays">--gcs-project-for-requester-pays</a><br />
					</td>
					<!--<td>String</td> -->
						<td></td>
					<td>Project to bill when accessing "requester pays" buckets. If unset, these buckets cannot be accessed.  User must have storage.buckets.get permission on the bucket being accessed.</td>
				</tr>
				<tr>
					<td><a href="#--gt-no-output">--gt-no-output</a><br />
					</td>
					<!--<td>boolean</td> -->
						<td>false</td>
					<td>do not generate output records</td>
				</tr>
				<tr>
					<td><a href="#--haplotype-output-padding-size">--haplotype-output-padding-size</a><br />
					</td>
					<!--<td>int</td> -->
						<td>8</td>
					<td>Number of N to append to best haplotype on output</td>
				</tr>
				<tr>
					<td><a href="#--help">--help</a><br />
								&nbsp;<em>-h</em>
					</td>
					<!--<td>boolean</td> -->
						<td>false</td>
					<td>display the help message</td>
				</tr>
				<tr>
					<td><a href="#--include-supp-align">--include-supp-align</a><br />
					</td>
					<!--<td>boolean</td> -->
						<td>false</td>
					<td>Include supplementary alignments</td>
				</tr>
				<tr>
					<td><a href="#--interval-merging-rule">--interval-merging-rule</a><br />
								&nbsp;<em>-imr</em>
					</td>
					<!--<td>IntervalMergingRule</td> -->
						<td>ALL</td>
					<td>Interval merging rule for abutting intervals</td>
				</tr>
				<tr>
					<td><a href="#--intervals">--intervals</a><br />
								&nbsp;<em>-L</em>
					</td>
					<!--<td>List[String]</td> -->
						<td></td>
					<td>One or more genomic intervals over which to operate</td>
				</tr>
				<tr>
					<td><a href="#--max-output-reads">--max-output-reads</a><br />
					</td>
					<!--<td>int</td> -->
						<td>20000000</td>
					<td>maximal number of reads to output</td>
				</tr>
				<tr>
					<td><a href="#--max-rq">--max-rq</a><br />
					</td>
					<!--<td>double</td> -->
						<td>0.0</td>
					<td>Maximal read quality</td>
				</tr>
				<tr>
					<td><a href="#--min-haplotype-score">--min-haplotype-score</a><br />
					</td>
					<!--<td>double</td> -->
						<td>0.0</td>
					<td>Minimal score (likelihood) on either haplotype</td>
				</tr>
				<tr>
					<td><a href="#--min-haplotype-score-delta">--min-haplotype-score-delta</a><br />
					</td>
					<!--<td>double</td> -->
						<td>0.0</td>
					<td>Minimal score (likelihood) delta between haplotypes</td>
				</tr>
				<tr>
					<td><a href="#--min-mq">--min-mq</a><br />
					</td>
					<!--<td>double</td> -->
						<td>0.0</td>
					<td>Minimal mapping quality</td>
				</tr>
				<tr>
					<td><a href="#--native-pair-hmm-threads">--native-pair-hmm-threads</a><br />
					</td>
					<!--<td>int</td> -->
						<td>4</td>
					<td>How many threads should a native pairHMM implementation use</td>
				</tr>
				<tr>
					<td><a href="#--native-pair-hmm-use-double-precision">--native-pair-hmm-use-double-precision</a><br />
					</td>
					<!--<td>boolean</td> -->
						<td>false</td>
					<td>use double precision in the native pairHmm. This is slower but matches the java implementation better</td>
				</tr>
				<tr>
					<td><a href="#--output-flow-length">--output-flow-length</a><br />
					</td>
					<!--<td>int</td> -->
						<td>0</td>
					<td>Required length of output flows</td>
				</tr>
				<tr>
					<td><a href="#--prepend-sequence">--prepend-sequence</a><br />
					</td>
					<!--<td>String</td> -->
						<td></td>
					<td>Sequence to prepend (barcode)</td>
				</tr>
				<tr>
					<td><a href="#--reference">--reference</a><br />
								&nbsp;<em>-R</em>
					</td>
					<!--<td>GATKPath</td> -->
						<td></td>
					<td>Reference sequence</td>
				</tr>
				<tr>
					<td><a href="#--sites-only-vcf-output">--sites-only-vcf-output</a><br />
					</td>
					<!--<td>boolean</td> -->
						<td>false</td>
					<td>If true, don't emit genotype fields when writing vcf file output.</td>
				</tr>
				<tr>
					<td><a href="#--subsampling-ratio">--subsampling-ratio</a><br />
					</td>
					<!--<td>double</td> -->
						<td>1.0</td>
					<td>subsampling ratio, should be between 0 and 1</td>
				</tr>
				<tr>
					<td><a href="#--version">--version</a><br />
					</td>
					<!--<td>boolean</td> -->
						<td>false</td>
					<td>display the version number for this tool</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Optional Common Arguments</th>
			</tr>
				<tr>
					<td><a href="#--add-output-sam-program-record">--add-output-sam-program-record</a><br />
					</td>
					<!--<td>boolean</td> -->
						<td>true</td>
					<td>If true, adds a PG tag to created SAM/BAM/CRAM files.</td>
				</tr>
				<tr>
					<td><a href="#--add-output-vcf-command-line">--add-output-vcf-command-line</a><br />
					</td>
					<!--<td>boolean</td> -->
						<td>true</td>
					<td>If true, adds a command line header line to created VCF files.</td>
				</tr>
				<tr>
					<td><a href="#--create-output-bam-index">--create-output-bam-index</a><br />
								&nbsp;<em>-OBI</em>
					</td>
					<!--<td>boolean</td> -->
						<td>true</td>
					<td>If true, create a BAM/CRAM index when writing a coordinate-sorted BAM/CRAM file.</td>
				</tr>
				<tr>
					<td><a href="#--create-output-bam-md5">--create-output-bam-md5</a><br />
								&nbsp;<em>-OBM</em>
					</td>
					<!--<td>boolean</td> -->
						<td>false</td>
					<td>If true, create a MD5 digest for any BAM/SAM/CRAM file created</td>
				</tr>
				<tr>
					<td><a href="#--create-output-variant-index">--create-output-variant-index</a><br />
								&nbsp;<em>-OVI</em>
					</td>
					<!--<td>boolean</td> -->
						<td>true</td>
					<td>If true, create a VCF index when writing a coordinate-sorted VCF file.</td>
				</tr>
				<tr>
					<td><a href="#--create-output-variant-md5">--create-output-variant-md5</a><br />
								&nbsp;<em>-OVM</em>
					</td>
					<!--<td>boolean</td> -->
						<td>false</td>
					<td>If true, create a a MD5 digest any VCF file created.</td>
				</tr>
				<tr>
					<td><a href="#--disable-read-filter">--disable-read-filter</a><br />
								&nbsp;<em>-DF</em>
					</td>
					<!--<td>List[String]</td> -->
						<td></td>
					<td>Read filters to be disabled before analysis</td>
				</tr>
				<tr>
					<td><a href="#--disable-tool-default-read-filters">--disable-tool-default-read-filters</a><br />
					</td>
					<!--<td>boolean</td> -->
						<td>false</td>
					<td>Disable all tool default read filters (WARNING: many tools will not function correctly without their default read filters on)</td>
				</tr>
				<tr>
					<td><a href="#--exclude-intervals">--exclude-intervals</a><br />
								&nbsp;<em>-XL</em>
					</td>
					<!--<td>List[String]</td> -->
						<td></td>
					<td>One or more genomic intervals to exclude from processing</td>
				</tr>
				<tr>
					<td><a href="#--gatk-config-file">--gatk-config-file</a><br />
					</td>
					<!--<td>String</td> -->
						<td></td>
					<td>A configuration file to use with the GATK.</td>
				</tr>
				<tr>
					<td><a href="#--interval-exclusion-padding">--interval-exclusion-padding</a><br />
								&nbsp;<em>-ixp</em>
					</td>
					<!--<td>int</td> -->
						<td>0</td>
					<td>Amount of padding (in bp) to add to each interval you are excluding.</td>
				</tr>
				<tr>
					<td><a href="#--interval-padding">--interval-padding</a><br />
								&nbsp;<em>-ip</em>
					</td>
					<!--<td>int</td> -->
						<td>0</td>
					<td>Amount of padding (in bp) to add to each interval you are including.</td>
				</tr>
				<tr>
					<td><a href="#--interval-set-rule">--interval-set-rule</a><br />
								&nbsp;<em>-isr</em>
					</td>
					<!--<td>IntervalSetRule</td> -->
						<td>UNION</td>
					<td>Set merging approach to use for combining interval inputs</td>
				</tr>
				<tr>
					<td><a href="#--lenient">--lenient</a><br />
								&nbsp;<em>-LE</em>
					</td>
					<!--<td>boolean</td> -->
						<td>false</td>
					<td>Lenient processing of VCF files</td>
				</tr>
				<tr>
					<td><a href="#--max-variants-per-shard">--max-variants-per-shard</a><br />
					</td>
					<!--<td>int</td> -->
						<td>0</td>
					<td>If non-zero, partitions VCF output into shards, each containing up to the given number of records.</td>
				</tr>
				<tr>
					<td><a href="#--QUIET">--QUIET</a><br />
					</td>
					<!--<td>Boolean</td> -->
						<td>false</td>
					<td>Whether to suppress job-summary info on System.err.</td>
				</tr>
				<tr>
					<td><a href="#--read-filter">--read-filter</a><br />
								&nbsp;<em>-RF</em>
					</td>
					<!--<td>List[String]</td> -->
						<td></td>
					<td>Read filters to be applied before analysis</td>
				</tr>
				<tr>
					<td><a href="#--read-index">--read-index</a><br />
					</td>
					<!--<td>List[GATKPath]</td> -->
						<td></td>
					<td>Indices to use for the read inputs. If specified, an index must be provided for every read input and in the same order as the read inputs. If this argument is not specified, the path to the index for each input will be inferred automatically.</td>
				</tr>
				<tr>
					<td><a href="#--read-validation-stringency">--read-validation-stringency</a><br />
								&nbsp;<em>-VS</em>
					</td>
					<!--<td>ValidationStringency</td> -->
						<td>SILENT</td>
					<td>Validation stringency for all SAM/BAM/CRAM/SRA files read by this program.  The default stringency value SILENT can improve performance when processing a BAM file in which variable-length data (read, qualities, tags) do not otherwise need to be decoded.</td>
				</tr>
				<tr>
					<td><a href="#--seconds-between-progress-updates">--seconds-between-progress-updates</a><br />
					</td>
					<!--<td>double</td> -->
						<td>10.0</td>
					<td>Output traversal statistics every time this many seconds elapse</td>
				</tr>
				<tr>
					<td><a href="#--sequence-dictionary">--sequence-dictionary</a><br />
					</td>
					<!--<td>GATKPath</td> -->
						<td></td>
					<td>Use the given sequence dictionary as the master/canonical sequence dictionary.  Must be a .dict file.</td>
				</tr>
				<tr>
					<td><a href="#--tmp-dir">--tmp-dir</a><br />
					</td>
					<!--<td>GATKPath</td> -->
						<td></td>
					<td>Temp directory to use.</td>
				</tr>
				<tr>
					<td><a href="#--use-jdk-deflater">--use-jdk-deflater</a><br />
								&nbsp;<em>-jdk-deflater</em>
					</td>
					<!--<td>boolean</td> -->
						<td>false</td>
					<td>Whether to use the JdkDeflater (as opposed to IntelDeflater)</td>
				</tr>
				<tr>
					<td><a href="#--use-jdk-inflater">--use-jdk-inflater</a><br />
								&nbsp;<em>-jdk-inflater</em>
					</td>
					<!--<td>boolean</td> -->
						<td>false</td>
					<td>Whether to use the JdkInflater (as opposed to IntelInflater)</td>
				</tr>
				<tr>
					<td><a href="#--verbosity">--verbosity</a><br />
					</td>
					<!--<td>LogLevel</td> -->
						<td>INFO</td>
					<td>Control verbosity of logging.</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Advanced Arguments</th>
			</tr>
				<tr>
					<td><a href="#--disable-cap-base-qualities-to-map-quality">--disable-cap-base-qualities-to-map-quality</a><br />
					</td>
					<!--<td>boolean</td> -->
						<td>false</td>
					<td>If false this disables capping of base qualities in the HMM to the mapping quality of the read</td>
				</tr>
				<tr>
					<td><a href="#--disable-symmetric-hmm-normalizing">--disable-symmetric-hmm-normalizing</a><br />
					</td>
					<!--<td>boolean</td> -->
						<td>false</td>
					<td>Toggle to revive legacy behavior of asymmetrically normalizing the arguments to the reference haplotype</td>
				</tr>
				<tr>
					<td><a href="#--expected-mismatch-rate-for-read-disqualification">--expected-mismatch-rate-for-read-disqualification</a><br />
					</td>
					<!--<td>double</td> -->
						<td>0.02</td>
					<td>Error rate used to set expectation for post HMM read disqualification based on mismatches</td>
				</tr>
				<tr>
					<td><a href="#--flow-disallow-probs-larger-than-call">--flow-disallow-probs-larger-than-call</a><br />
					</td>
					<!--<td>boolean</td> -->
						<td>false</td>
					<td>Cap probabilities of error to 1 relative to base call</td>
				</tr>
				<tr>
					<td><a href="#--flow-fill-empty-bins-value">--flow-fill-empty-bins-value</a><br />
					</td>
					<!--<td>double</td> -->
						<td>0.001</td>
					<td>Value to fill the zeros of the matrix with</td>
				</tr>
				<tr>
					<td><a href="#--flow-lump-probs">--flow-lump-probs</a><br />
					</td>
					<!--<td>boolean</td> -->
						<td>false</td>
					<td>Should all probabilities of insertion or deletion in the flow be combined together</td>
				</tr>
				<tr>
					<td><a href="#--flow-matrix-mods">--flow-matrix-mods</a><br />
					</td>
					<!--<td>String</td> -->
						<td></td>
					<td>Modifications instructions to the read flow matrix. Format is src,dst{,src,dst}+. Example: 10,12,11,12 - these instructions will copy element 10 into 11 and 12</td>
				</tr>
				<tr>
					<td><a href="#--flow-probability-scaling-factor">--flow-probability-scaling-factor</a><br />
					</td>
					<!--<td>int</td> -->
						<td>10</td>
					<td>probability scaling factor for (phred=10) for probability quantization</td>
				</tr>
				<tr>
					<td><a href="#--flow-probability-threshold">--flow-probability-threshold</a><br />
					</td>
					<!--<td>double</td> -->
						<td>0.003</td>
					<td>Lowest probability ratio to be used as an option</td>
				</tr>
				<tr>
					<td><a href="#--flow-quantization-bins">--flow-quantization-bins</a><br />
					</td>
					<!--<td>int</td> -->
						<td>121</td>
					<td>Number of bins for probability quantization</td>
				</tr>
				<tr>
					<td><a href="#--flow-remove-non-single-base-pair-indels">--flow-remove-non-single-base-pair-indels</a><br />
					</td>
					<!--<td>boolean</td> -->
						<td>false</td>
					<td>Should the probabilities of more then 1 indel be used</td>
				</tr>
				<tr>
					<td><a href="#--flow-remove-one-zero-probs">--flow-remove-one-zero-probs</a><br />
					</td>
					<!--<td>boolean</td> -->
						<td>false</td>
					<td>Remove probabilities of basecall of zero from non-zero genome</td>
				</tr>
				<tr>
					<td><a href="#--flow-report-insertion-or-deletion">--flow-report-insertion-or-deletion</a><br />
					</td>
					<!--<td>boolean</td> -->
						<td>false</td>
					<td>Report either insertion or deletion, probability, not both</td>
				</tr>
				<tr>
					<td><a href="#--flow-retain-max-n-probs-base-format">--flow-retain-max-n-probs-base-format</a><br />
					</td>
					<!--<td>boolean</td> -->
						<td>false</td>
					<td>Keep only hmer/2 probabilities (like in base format)</td>
				</tr>
				<tr>
					<td><a href="#--flow-symmetric-indel-probs">--flow-symmetric-indel-probs</a><br />
					</td>
					<!--<td>boolean</td> -->
						<td>false</td>
					<td>Should indel probabilities be symmetric in flow</td>
				</tr>
				<tr>
					<td><a href="#--flow-use-t0-tag">--flow-use-t0-tag</a><br />
					</td>
					<!--<td>boolean</td> -->
						<td>false</td>
					<td>Use t0 tag if exists in the read to create flow matrix</td>
				</tr>
				<tr>
					<td><a href="#--keep-boundary-flows">--keep-boundary-flows</a><br />
					</td>
					<!--<td>boolean</td> -->
						<td>false</td>
					<td>prevent spreading of boundary flows.</td>
				</tr>
				<tr>
					<td><a href="#--likelihood-calculation-engine">--likelihood-calculation-engine</a><br />
					</td>
					<!--<td>Implementation</td> -->
						<td>PairHMM</td>
					<td>What likelihood calculation engine to use to calculate the relative likelihood of reads vs haplotypes</td>
				</tr>
				<tr>
					<td><a href="#--pair-hmm-gap-continuation-penalty">--pair-hmm-gap-continuation-penalty</a><br />
					</td>
					<!--<td>int</td> -->
						<td>10</td>
					<td>Flat gap continuation penalty for use in the Pair HMM</td>
				</tr>
				<tr>
					<td><a href="#--pair-hmm-implementation">--pair-hmm-implementation</a><br />
								&nbsp;<em>-pairHMM</em>
					</td>
					<!--<td>Implementation</td> -->
						<td>FASTEST_AVAILABLE</td>
					<td>The PairHMM implementation to use for genotype likelihood calculations</td>
				</tr>
				<tr>
					<td><a href="#--pair-hmm-results-file">--pair-hmm-results-file</a><br />
					</td>
					<!--<td>GATKPath</td> -->
						<td></td>
					<td>File to write exact pairHMM inputs/outputs to for debugging purposes</td>
				</tr>
				<tr>
					<td><a href="#--pcr-indel-model">--pcr-indel-model</a><br />
					</td>
					<!--<td>PCRErrorModel</td> -->
						<td>CONSERVATIVE</td>
					<td>The PCR indel model to use</td>
				</tr>
				<tr>
					<td><a href="#--phred-scaled-global-read-mismapping-rate">--phred-scaled-global-read-mismapping-rate</a><br />
					</td>
					<!--<td>int</td> -->
						<td>45</td>
					<td>The global assumed mismapping rate for reads</td>
				</tr>
				<tr>
					<td><a href="#--showHidden">--showHidden</a><br />
					</td>
					<!--<td>boolean</td> -->
						<td>false</td>
					<td>display hidden arguments</td>
				</tr>
					</tbody>
				</table>

					<h3>Argument details</h3>
					<p>Arguments in this list are specific to this tool. Keep in mind that other arguments are available that are shared with other tools (e.g. command-line GATK arguments); see Inherited arguments above.</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--add-output-sam-program-record">--add-output-sam-program-record </a>
			 / <small>-add-output-sam-program-record</small>
		</h3>
		<p class="args">
			<b>If true, adds a PG tag to created SAM/BAM/CRAM files.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">true</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--add-output-vcf-command-line">--add-output-vcf-command-line </a>
			 / <small>-add-output-vcf-command-line</small>
		</h3>
		<p class="args">
			<b>If true, adds a command line header line to created VCF files.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">true</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--ancestral-translators-base-path">--ancestral-translators-base-path </a>
			
		</h3>
		<p class="args">
			<b>base path for ancestral translation ancestral.contig.csv files</b><br />
			
		</p>
		<p>
				<span class="badge badge-important">R</span>
		<span class="label label-info ">GATKPath</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--append-sequence">--append-sequence </a>
			
		</h3>
		<p class="args">
			<b>Sequence to append (adapter)</b><br />
			
		</p>
		<p>
		<span class="label label-info ">String</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--arguments_file">--arguments_file </a>
			
		</h3>
		<p class="args">
			<b>read one or more arguments files and add them to the command line</b><br />
			
		</p>
		<p>
		<span class="label label-info ">List[File]</span>
			&nbsp;<span class="label">[]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--base-quality-score-threshold">--base-quality-score-threshold </a>
			
		</h3>
		<p class="args">
			<b>Base qualities below this threshold will be reduced to the minimum (6)</b><br />
			Bases with a quality below this threshold will reduced to the minimum usable qualiy score (6).
		</p>
		<p>
		<span class="label label-info ">byte</span>
			&nbsp;<span class="label">18</span>
			&nbsp;<span class="label label-warning">[ [ -</span>
			&nbsp;<span class="label label-warning"> ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--cloud-index-prefetch-buffer">--cloud-index-prefetch-buffer </a>
			 / <small>-CIPB</small>
		</h3>
		<p class="args">
			<b>Size of the cloud-only prefetch buffer (in MB; 0 to disable). Defaults to cloudPrefetchBuffer if unset.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">int</span>
			&nbsp;<span class="label">-1</span>
			&nbsp;<span class="label label-warning">[ [ -</span>
			&nbsp;<span class="label label-warning"> ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--cloud-prefetch-buffer">--cloud-prefetch-buffer </a>
			 / <small>-CPB</small>
		</h3>
		<p class="args">
			<b>Size of the cloud-only prefetch buffer (in MB; 0 to disable).</b><br />
			
		</p>
		<p>
		<span class="label label-info ">int</span>
			&nbsp;<span class="label">40</span>
			&nbsp;<span class="label label-warning">[ [ -</span>
			&nbsp;<span class="label label-warning"> ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--create-output-bam-index">--create-output-bam-index </a>
			 / <small>-OBI</small>
		</h3>
		<p class="args">
			<b>If true, create a BAM/CRAM index when writing a coordinate-sorted BAM/CRAM file.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">true</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--create-output-bam-md5">--create-output-bam-md5 </a>
			 / <small>-OBM</small>
		</h3>
		<p class="args">
			<b>If true, create a MD5 digest for any BAM/SAM/CRAM file created</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--create-output-variant-index">--create-output-variant-index </a>
			 / <small>-OVI</small>
		</h3>
		<p class="args">
			<b>If true, create a VCF index when writing a coordinate-sorted VCF file.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">true</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--create-output-variant-md5">--create-output-variant-md5 </a>
			 / <small>-OVM</small>
		</h3>
		<p class="args">
			<b>If true, create a a MD5 digest any VCF file created.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--disable-bam-index-caching">--disable-bam-index-caching </a>
			 / <small>-DBIC</small>
		</h3>
		<p class="args">
			<b>If true, don't cache bam indexes, this will reduce memory requirements but may harm performance if many intervals are specified.  Caching is automatically disabled if there are no intervals specified.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--disable-cap-base-qualities-to-map-quality">--disable-cap-base-qualities-to-map-quality </a>
			
		</h3>
		<p class="args">
			<b>If false this disables capping of base qualities in the HMM to the mapping quality of the read</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--disable-read-filter">--disable-read-filter </a>
			 / <small>-DF</small>
		</h3>
		<p class="args">
			<b>Read filters to be disabled before analysis</b><br />
			
		</p>
		<p>
		<span class="label label-info ">List[String]</span>
			&nbsp;<span class="label">[]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--disable-sequence-dictionary-validation">--disable-sequence-dictionary-validation </a>
			 / <small>-disable-sequence-dictionary-validation</small>
		</h3>
		<p class="args">
			<b>If specified, do not check the sequence dictionaries from our inputs for compatibility. Use at your own risk!</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--disable-symmetric-hmm-normalizing">--disable-symmetric-hmm-normalizing </a>
			
		</h3>
		<p class="args">
			<b>Toggle to revive legacy behavior of asymmetrically normalizing the arguments to the reference haplotype</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--disable-tool-default-read-filters">--disable-tool-default-read-filters </a>
			 / <small>-disable-tool-default-read-filters</small>
		</h3>
		<p class="args">
			<b>Disable all tool default read filters (WARNING: many tools will not function correctly without their default read filters on)</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--discard-non-polyt-softclipped-reads">--discard-non-polyt-softclipped-reads </a>
			
		</h3>
		<p class="args">
			<b>Discard reads which are softclipped, unless the softclip is polyT, defaults to true</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--dont-use-dragstr-pair-hmm-scores">--dont-use-dragstr-pair-hmm-scores </a>
			
		</h3>
		<p class="args">
			<b>disable DRAGstr pair-hmm score even when dragstr-params-path was provided</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--dragstr-het-hom-ratio">--dragstr-het-hom-ratio </a>
			
		</h3>
		<p class="args">
			<b>het to hom prior ratio use with DRAGstr on</b><br />
			
		</p>
		<p>
		<span class="label label-info ">int</span>
			&nbsp;<span class="label">2</span>
			&nbsp;<span class="label label-warning">[ [ -</span>
			&nbsp;<span class="label label-warning"> ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--dragstr-params-path">--dragstr-params-path </a>
			
		</h3>
		<p class="args">
			<b>location of the DRAGstr model parameters for STR error correction used in the Pair HMM. When provided, it overrides other PCR error correcting mechanisms</b><br />
			
		</p>
		<p>
		<span class="label label-info ">GATKPath</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--enable-dynamic-read-disqualification-for-genotyping">--enable-dynamic-read-disqualification-for-genotyping </a>
			
		</h3>
		<p class="args">
			<b>Will enable less strict read disqualification low base quality reads</b><br />
			If enabled, rather than disqualifying all reads over a threshold of minimum hmm scores we will instead choose a less strict
 and less aggressive cap for disqualification based on the read length and base qualities.
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--exclude-intervals">--exclude-intervals </a>
			 / <small>-XL</small>
		</h3>
		<p class="args">
			<b>One or more genomic intervals to exclude from processing</b><br />
			Use this argument to exclude certain parts of the genome from the analysis (like -L, but the opposite).
 This argument can be specified multiple times. You can use samtools-style intervals either explicitly on the
 command line (e.g. -XL 1 or -XL 1:100-200) or by loading in a file containing a list of intervals
 (e.g. -XL myFile.intervals).
		</p>
		<p>
		<span class="label label-info ">List[String]</span>
			&nbsp;<span class="label">[]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--expected-mismatch-rate-for-read-disqualification">--expected-mismatch-rate-for-read-disqualification </a>
			
		</h3>
		<p class="args">
			<b>Error rate used to set expectation for post HMM read disqualification based on mismatches</b><br />
			
		</p>
		<p>
		<span class="label label-info ">double</span>
			&nbsp;<span class="label">0.02</span>
			&nbsp;<span class="label label-warning">[ [ -</span>
			&nbsp;<span class="label label-warning"> ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--false-snp-compensation">--false-snp-compensation </a>
			
		</h3>
		<p class="args">
			<b>skip haplotype bases until same base as read starts (false SNP compensation)</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--fill-softclipped-reads">--fill-softclipped-reads </a>
			
		</h3>
		<p class="args">
			<b>Softclipped reads should be filled from haplotype, otherwise (default) filled with -83</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--fill-trimmed-reads">--fill-trimmed-reads </a>
			
		</h3>
		<p class="args">
			<b>Reads with tm:Q or tm:Z should be filled from haplotype, otherwise (default) filled with -80</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--fill-trimmed-reads-Q">--fill-trimmed-reads-Q </a>
			
		</h3>
		<p class="args">
			<b>Reads with tm:Q should be filled from haplotype, otherwise (default) filled with -80</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--fill-trimmed-reads-Z">--fill-trimmed-reads-Z </a>
			
		</h3>
		<p class="args">
			<b>Reads with tm:Z should be filled from haplotype, otherwise (default) filled with -80</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--flow-disallow-probs-larger-than-call">--flow-disallow-probs-larger-than-call </a>
			
		</h3>
		<p class="args">
			<b>Cap probabilities of error to 1 relative to base call</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--flow-fill-empty-bins-value">--flow-fill-empty-bins-value </a>
			
		</h3>
		<p class="args">
			<b>Value to fill the zeros of the matrix with</b><br />
			
		</p>
		<p>
		<span class="label label-info ">double</span>
			&nbsp;<span class="label">0.001</span>
			&nbsp;<span class="label label-warning">[ [ -</span>
			&nbsp;<span class="label label-warning"> ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--flow-lump-probs">--flow-lump-probs </a>
			
		</h3>
		<p class="args">
			<b>Should all probabilities of insertion or deletion in the flow be combined together</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--flow-matrix-mods">--flow-matrix-mods </a>
			
		</h3>
		<p class="args">
			<b>Modifications instructions to the read flow matrix. Format is src,dst{,src,dst}+. Example: 10,12,11,12 - these instructions will copy element 10 into 11 and 12</b><br />
			
		</p>
		<p>
		<span class="label label-info ">String</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--flow-probability-scaling-factor">--flow-probability-scaling-factor </a>
			
		</h3>
		<p class="args">
			<b>probability scaling factor for (phred=10) for probability quantization</b><br />
			
		</p>
		<p>
		<span class="label label-info ">int</span>
			&nbsp;<span class="label">10</span>
			&nbsp;<span class="label label-warning">[ [ -</span>
			&nbsp;<span class="label label-warning"> ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--flow-probability-threshold">--flow-probability-threshold </a>
			
		</h3>
		<p class="args">
			<b>Lowest probability ratio to be used as an option</b><br />
			
		</p>
		<p>
		<span class="label label-info ">double</span>
			&nbsp;<span class="label">0.003</span>
			&nbsp;<span class="label label-warning">[ [ -</span>
			&nbsp;<span class="label label-warning"> ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--flow-quantization-bins">--flow-quantization-bins </a>
			
		</h3>
		<p class="args">
			<b>Number of bins for probability quantization</b><br />
			
		</p>
		<p>
		<span class="label label-info ">int</span>
			&nbsp;<span class="label">121</span>
			&nbsp;<span class="label label-warning">[ [ -</span>
			&nbsp;<span class="label label-warning"> ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--flow-remove-non-single-base-pair-indels">--flow-remove-non-single-base-pair-indels </a>
			
		</h3>
		<p class="args">
			<b>Should the probabilities of more then 1 indel be used</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--flow-remove-one-zero-probs">--flow-remove-one-zero-probs </a>
			
		</h3>
		<p class="args">
			<b>Remove probabilities of basecall of zero from non-zero genome</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--flow-report-insertion-or-deletion">--flow-report-insertion-or-deletion </a>
			
		</h3>
		<p class="args">
			<b>Report either insertion or deletion, probability, not both</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--flow-retain-max-n-probs-base-format">--flow-retain-max-n-probs-base-format </a>
			
		</h3>
		<p class="args">
			<b>Keep only hmer/2 probabilities (like in base format)</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--flow-symmetric-indel-probs">--flow-symmetric-indel-probs </a>
			
		</h3>
		<p class="args">
			<b>Should indel probabilities be symmetric in flow</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--flow-use-t0-tag">--flow-use-t0-tag </a>
			
		</h3>
		<p class="args">
			<b>Use t0 tag if exists in the read to create flow matrix</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--gatk-config-file">--gatk-config-file </a>
			
		</h3>
		<p class="args">
			<b>A configuration file to use with the GATK.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">String</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--gcs-max-retries">--gcs-max-retries </a>
			 / <small>-gcs-retries</small>
		</h3>
		<p class="args">
			<b>If the GCS bucket channel errors out, how many times it will attempt to re-initiate the connection</b><br />
			
		</p>
		<p>
		<span class="label label-info ">int</span>
			&nbsp;<span class="label">20</span>
			&nbsp;<span class="label label-warning">[ [ -</span>
			&nbsp;<span class="label label-warning"> ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--gcs-project-for-requester-pays">--gcs-project-for-requester-pays </a>
			
		</h3>
		<p class="args">
			<b>Project to bill when accessing "requester pays" buckets. If unset, these buckets cannot be accessed.  User must have storage.buckets.get permission on the bucket being accessed.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">String</span>
			&nbsp;<span class="label">""</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--gt-no-output">--gt-no-output </a>
			
		</h3>
		<p class="args">
			<b>do not generate output records</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--haplotype-output-padding-size">--haplotype-output-padding-size </a>
			
		</h3>
		<p class="args">
			<b>Number of N to append to best haplotype on output</b><br />
			
		</p>
		<p>
		<span class="label label-info ">int</span>
			&nbsp;<span class="label">8</span>
			&nbsp;<span class="label label-warning">[ [ -</span>
			&nbsp;<span class="label label-warning"> ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--help">--help </a>
			 / <small>-h</small>
		</h3>
		<p class="args">
			<b>display the help message</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--include-supp-align">--include-supp-align </a>
			
		</h3>
		<p class="args">
			<b>Include supplementary alignments</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--input">--input </a>
			 / <small>-I</small>
		</h3>
		<p class="args">
			<b>BAM/SAM/CRAM file containing reads</b><br />
			
		</p>
		<p>
				<span class="badge badge-important">R</span>
		<span class="label label-info ">List[GATKPath]</span>
			&nbsp;<span class="label">[]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--interval-exclusion-padding">--interval-exclusion-padding </a>
			 / <small>-ixp</small>
		</h3>
		<p class="args">
			<b>Amount of padding (in bp) to add to each interval you are excluding.</b><br />
			Use this to add padding to the intervals specified using -XL. For example, '-XL 1:100' with a
 padding value of 20 would turn into '-XL 1:80-120'. This is typically used to add padding around targets when
 analyzing exomes.
		</p>
		<p>
		<span class="label label-info ">int</span>
			&nbsp;<span class="label">0</span>
			&nbsp;<span class="label label-warning">[ [ -</span>
			&nbsp;<span class="label label-warning"> ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--interval-merging-rule">--interval-merging-rule </a>
			 / <small>-imr</small>
		</h3>
		<p class="args">
			<b>Interval merging rule for abutting intervals</b><br />
			By default, the program merges abutting intervals (i.e. intervals that are directly side-by-side but do not
 actually overlap) into a single continuous interval. However you can change this behavior if you want them to be
 treated as separate intervals instead.
		</p>
			<p>
				The --interval-merging-rule argument is an enumerated type (IntervalMergingRule), which can have one of the following values:
			<dl class="enum">
					<dt class="enum">ALL</dt>
					<dd class="enum"></dd>
					<dt class="enum">OVERLAPPING_ONLY</dt>
					<dd class="enum"></dd>
			</dl>
			</p>
		<p>
		<span class="label label-info ">IntervalMergingRule</span>
			&nbsp;<span class="label">ALL</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--interval-padding">--interval-padding </a>
			 / <small>-ip</small>
		</h3>
		<p class="args">
			<b>Amount of padding (in bp) to add to each interval you are including.</b><br />
			Use this to add padding to the intervals specified using -L. For example, '-L 1:100' with a
 padding value of 20 would turn into '-L 1:80-120'. This is typically used to add padding around targets when
 analyzing exomes.
		</p>
		<p>
		<span class="label label-info ">int</span>
			&nbsp;<span class="label">0</span>
			&nbsp;<span class="label label-warning">[ [ -</span>
			&nbsp;<span class="label label-warning"> ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--interval-set-rule">--interval-set-rule </a>
			 / <small>-isr</small>
		</h3>
		<p class="args">
			<b>Set merging approach to use for combining interval inputs</b><br />
			By default, the program will take the UNION of all intervals specified using -L and/or -XL. However, you can
 change this setting for -L, for example if you want to take the INTERSECTION of the sets instead. E.g. to
 perform the analysis only on chromosome 1 exomes, you could specify -L exomes.intervals -L 1 --interval-set-rule
 INTERSECTION. However, it is not possible to modify the merging approach for intervals passed using -XL (they will
 always be merged using UNION).

 Note that if you specify both -L and -XL, the -XL interval set will be subtracted from the -L interval set.
		</p>
			<p>
				The --interval-set-rule argument is an enumerated type (IntervalSetRule), which can have one of the following values:
			<dl class="enum">
					<dt class="enum">UNION</dt>
					<dd class="enum">Take the union of all intervals</dd>
					<dt class="enum">INTERSECTION</dt>
					<dd class="enum">Take the intersection of intervals (the subset that overlaps all intervals specified)</dd>
			</dl>
			</p>
		<p>
		<span class="label label-info ">IntervalSetRule</span>
			&nbsp;<span class="label">UNION</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--intervals">--intervals </a>
			 / <small>-L</small>
		</h3>
		<p class="args">
			<b>One or more genomic intervals over which to operate</b><br />
			
		</p>
		<p>
		<span class="label label-info ">List[String]</span>
			&nbsp;<span class="label">[]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--keep-boundary-flows">--keep-boundary-flows </a>
			
		</h3>
		<p class="args">
			<b>prevent spreading of boundary flows.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--lenient">--lenient </a>
			 / <small>-LE</small>
		</h3>
		<p class="args">
			<b>Lenient processing of VCF files</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--likelihood-calculation-engine">--likelihood-calculation-engine </a>
			
		</h3>
		<p class="args">
			<b>What likelihood calculation engine to use to calculate the relative likelihood of reads vs haplotypes</b><br />
			
		</p>
			<p>
				The --likelihood-calculation-engine argument is an enumerated type (Implementation), which can have one of the following values:
			<dl class="enum">
					<dt class="enum">PairHMM</dt>
					<dd class="enum">Classic full pair-hmm all haplotypes vs all reads.</dd>
					<dt class="enum">FlowBased</dt>
					<dd class="enum"></dd>
					<dt class="enum">FlowBasedHMM</dt>
					<dd class="enum"></dd>
			</dl>
			</p>
		<p>
		<span class="label label-info ">Implementation</span>
			&nbsp;<span class="label">PairHMM</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--maternal-ref">--maternal-ref </a>
			
		</h3>
		<p class="args">
			<b>maternal reference file</b><br />
			
		</p>
		<p>
				<span class="badge badge-important">R</span>
		<span class="label label-info ">GATKPath</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--max-output-reads">--max-output-reads </a>
			
		</h3>
		<p class="args">
			<b>maximal number of reads to output</b><br />
			
		</p>
		<p>
		<span class="label label-info ">int</span>
			&nbsp;<span class="label">20000000</span>
			&nbsp;<span class="label label-warning">[ [ -</span>
			&nbsp;<span class="label label-warning"> ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--max-rq">--max-rq </a>
			
		</h3>
		<p class="args">
			<b>Maximal read quality</b><br />
			
		</p>
		<p>
		<span class="label label-info ">double</span>
			&nbsp;<span class="label">0.0</span>
			&nbsp;<span class="label label-warning">[ [ -</span>
			&nbsp;<span class="label label-warning"> ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--max-variants-per-shard">--max-variants-per-shard </a>
			
		</h3>
		<p class="args">
			<b>If non-zero, partitions VCF output into shards, each containing up to the given number of records.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">int</span>
			&nbsp;<span class="label">0</span>
			&nbsp;<span class="label label-warning">[ [ 0</span>
			&nbsp;<span class="label label-warning"> ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--min-haplotype-score">--min-haplotype-score </a>
			
		</h3>
		<p class="args">
			<b>Minimal score (likelihood) on either haplotype</b><br />
			
		</p>
		<p>
		<span class="label label-info ">double</span>
			&nbsp;<span class="label">0.0</span>
			&nbsp;<span class="label label-warning">[ [ -</span>
			&nbsp;<span class="label label-warning"> ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--min-haplotype-score-delta">--min-haplotype-score-delta </a>
			
		</h3>
		<p class="args">
			<b>Minimal score (likelihood) delta between haplotypes</b><br />
			
		</p>
		<p>
		<span class="label label-info ">double</span>
			&nbsp;<span class="label">0.0</span>
			&nbsp;<span class="label label-warning">[ [ -</span>
			&nbsp;<span class="label label-warning"> ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--min-mq">--min-mq </a>
			
		</h3>
		<p class="args">
			<b>Minimal mapping quality</b><br />
			
		</p>
		<p>
		<span class="label label-info ">double</span>
			&nbsp;<span class="label">0.0</span>
			&nbsp;<span class="label label-warning">[ [ -</span>
			&nbsp;<span class="label label-warning"> ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--native-pair-hmm-threads">--native-pair-hmm-threads </a>
			
		</h3>
		<p class="args">
			<b>How many threads should a native pairHMM implementation use</b><br />
			
		</p>
		<p>
		<span class="label label-info ">int</span>
			&nbsp;<span class="label">4</span>
			&nbsp;<span class="label label-warning">[ [ -</span>
			&nbsp;<span class="label label-warning"> ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--native-pair-hmm-use-double-precision">--native-pair-hmm-use-double-precision </a>
			
		</h3>
		<p class="args">
			<b>use double precision in the native pairHmm. This is slower but matches the java implementation better</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--output-csv">--output-csv </a>
			
		</h3>
		<p class="args">
			<b>main CSV output file. the file containing maternal/parental maternal and paternal haplotype sequences and scores (and many more columns). supported file extensions: .csv, .csv.gz.</b><br />
			
		</p>
		<p>
				<span class="badge badge-important">R</span>
		<span class="label label-info ">GATKPath</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--output-flow-length">--output-flow-length </a>
			
		</h3>
		<p class="args">
			<b>Required length of output flows</b><br />
			
		</p>
		<p>
		<span class="label label-info ">int</span>
			&nbsp;<span class="label">0</span>
			&nbsp;<span class="label label-warning">[ [ -</span>
			&nbsp;<span class="label label-warning"> ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--pair-hmm-gap-continuation-penalty">--pair-hmm-gap-continuation-penalty </a>
			
		</h3>
		<p class="args">
			<b>Flat gap continuation penalty for use in the Pair HMM</b><br />
			
		</p>
		<p>
		<span class="label label-info ">int</span>
			&nbsp;<span class="label">10</span>
			&nbsp;<span class="label label-warning">[ [ -</span>
			&nbsp;<span class="label label-warning"> ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--pair-hmm-implementation">--pair-hmm-implementation </a>
			 / <small>-pairHMM</small>
		</h3>
		<p class="args">
			<b>The PairHMM implementation to use for genotype likelihood calculations</b><br />
			The PairHMM implementation to use for genotype likelihood calculations. The various implementations balance a tradeoff of accuracy and runtime.
		</p>
			<p>
				The --pair-hmm-implementation argument is an enumerated type (Implementation), which can have one of the following values:
			<dl class="enum">
					<dt class="enum">EXACT</dt>
					<dd class="enum"></dd>
					<dt class="enum">ORIGINAL</dt>
					<dd class="enum"></dd>
					<dt class="enum">LOGLESS_CACHING</dt>
					<dd class="enum"></dd>
					<dt class="enum">AVX_LOGLESS_CACHING</dt>
					<dd class="enum"></dd>
					<dt class="enum">AVX_LOGLESS_CACHING_OMP</dt>
					<dd class="enum"></dd>
					<dt class="enum">FASTEST_AVAILABLE</dt>
					<dd class="enum"></dd>
			</dl>
			</p>
		<p>
		<span class="label label-info ">Implementation</span>
			&nbsp;<span class="label">FASTEST_AVAILABLE</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--pair-hmm-results-file">--pair-hmm-results-file </a>
			
		</h3>
		<p class="args">
			<b>File to write exact pairHMM inputs/outputs to for debugging purposes</b><br />
			Argument for generating a file of all of the inputs and outputs for the pair hmm
		</p>
		<p>
		<span class="label label-info ">GATKPath</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--paternal-ref">--paternal-ref </a>
			
		</h3>
		<p class="args">
			<b>paternal reference file</b><br />
			
		</p>
		<p>
				<span class="badge badge-important">R</span>
		<span class="label label-info ">GATKPath</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--pcr-indel-model">--pcr-indel-model </a>
			
		</h3>
		<p class="args">
			<b>The PCR indel model to use</b><br />
			When calculating the likelihood of variants, we can try to correct for PCR errors that cause indel artifacts.
 The correction is based on the reference context, and acts specifically around repetitive sequences that tend
 to cause PCR errors). The variant likelihoods are penalized in increasing scale as the context around a
 putative indel is more repetitive (e.g. long homopolymer). The correction can be disabling by specifying
 '-pcrModel NONE'; in that case the default base insertion/deletion qualities will be used (or taken from the
 read if generated through the BaseRecalibrator). <b>VERY IMPORTANT: when using PCR-free sequencing data we
 definitely recommend setting this argument to NONE</b>.
		</p>
			<p>
				The --pcr-indel-model argument is an enumerated type (PCRErrorModel), which can have one of the following values:
			<dl class="enum">
					<dt class="enum">NONE</dt>
					<dd class="enum">no specialized PCR error model will be applied; if base insertion/deletion qualities are present they will be used</dd>
					<dt class="enum">HOSTILE</dt>
					<dd class="enum">a most aggressive model will be applied that sacrifices true positives in order to remove more false positives</dd>
					<dt class="enum">AGGRESSIVE</dt>
					<dd class="enum">a more aggressive model will be applied that sacrifices true positives in order to remove more false positives</dd>
					<dt class="enum">CONSERVATIVE</dt>
					<dd class="enum">a less aggressive model will be applied that tries to maintain a high true positive rate at the expense of allowing more false positives</dd>
			</dl>
			</p>
		<p>
		<span class="label label-info ">PCRErrorModel</span>
			&nbsp;<span class="label">CONSERVATIVE</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--phred-scaled-global-read-mismapping-rate">--phred-scaled-global-read-mismapping-rate </a>
			
		</h3>
		<p class="args">
			<b>The global assumed mismapping rate for reads</b><br />
			The phredScaledGlobalReadMismappingRate reflects the average global mismapping rate of all reads, regardless of their
 mapping quality.  This term effects the probability that a read originated from the reference haplotype, regardless of
 its edit distance from the reference, in that the read could have originated from the reference haplotype but
 from another location in the genome.  Suppose a read has many mismatches from the reference, say like 5, but
 has a very high mapping quality of 60.  Without this parameter, the read would contribute 5 * Q30 evidence
 in favor of its 5 mismatch haplotype compared to reference, potentially enough to make a call off that single
 read for all of these events.  With this parameter set to Q30, though, the maximum evidence against any haplotype
 that this (and any) read could contribute is Q30.

 Set this term to any negative number to turn off the global mapping rate.
		</p>
		<p>
		<span class="label label-info ">int</span>
			&nbsp;<span class="label">45</span>
			&nbsp;<span class="label label-warning">[ [ -</span>
			&nbsp;<span class="label label-warning"> ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--prepend-sequence">--prepend-sequence </a>
			
		</h3>
		<p class="args">
			<b>Sequence to prepend (barcode)</b><br />
			
		</p>
		<p>
		<span class="label label-info ">String</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--QUIET">--QUIET </a>
			
		</h3>
		<p class="args">
			<b>Whether to suppress job-summary info on System.err.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">Boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--read-filter">--read-filter </a>
			 / <small>-RF</small>
		</h3>
		<p class="args">
			<b>Read filters to be applied before analysis</b><br />
			
		</p>
		<p>
		<span class="label label-info ">List[String]</span>
			&nbsp;<span class="label">[]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--read-index">--read-index </a>
			 / <small>-read-index</small>
		</h3>
		<p class="args">
			<b>Indices to use for the read inputs. If specified, an index must be provided for every read input and in the same order as the read inputs. If this argument is not specified, the path to the index for each input will be inferred automatically.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">List[GATKPath]</span>
			&nbsp;<span class="label">[]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--read-validation-stringency">--read-validation-stringency </a>
			 / <small>-VS</small>
		</h3>
		<p class="args">
			<b>Validation stringency for all SAM/BAM/CRAM/SRA files read by this program.  The default stringency value SILENT can improve performance when processing a BAM file in which variable-length data (read, qualities, tags) do not otherwise need to be decoded.</b><br />
			
		</p>
			<p>
				The --read-validation-stringency argument is an enumerated type (ValidationStringency), which can have one of the following values:
			<dl class="enum">
					<dt class="enum">STRICT</dt>
					<dd class="enum"></dd>
					<dt class="enum">LENIENT</dt>
					<dd class="enum"></dd>
					<dt class="enum">SILENT</dt>
					<dd class="enum"></dd>
			</dl>
			</p>
		<p>
		<span class="label label-info ">ValidationStringency</span>
			&nbsp;<span class="label">SILENT</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--reference">--reference </a>
			 / <small>-R</small>
		</h3>
		<p class="args">
			<b>Reference sequence</b><br />
			
		</p>
		<p>
		<span class="label label-info ">GATKPath</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--seconds-between-progress-updates">--seconds-between-progress-updates </a>
			 / <small>-seconds-between-progress-updates</small>
		</h3>
		<p class="args">
			<b>Output traversal statistics every time this many seconds elapse</b><br />
			
		</p>
		<p>
		<span class="label label-info ">double</span>
			&nbsp;<span class="label">10.0</span>
			&nbsp;<span class="label label-warning">[ [ -</span>
			&nbsp;<span class="label label-warning"> ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--sequence-dictionary">--sequence-dictionary </a>
			 / <small>-sequence-dictionary</small>
		</h3>
		<p class="args">
			<b>Use the given sequence dictionary as the master/canonical sequence dictionary.  Must be a .dict file.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">GATKPath</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--showHidden">--showHidden </a>
			 / <small>-showHidden</small>
		</h3>
		<p class="args">
			<b>display hidden arguments</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--sites-only-vcf-output">--sites-only-vcf-output </a>
			
		</h3>
		<p class="args">
			<b>If true, don't emit genotype fields when writing vcf file output.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--subsampling-ratio">--subsampling-ratio </a>
			
		</h3>
		<p class="args">
			<b>subsampling ratio, should be between 0 and 1</b><br />
			
		</p>
		<p>
		<span class="label label-info ">double</span>
			&nbsp;<span class="label">1.0</span>
			&nbsp;<span class="label label-warning">[ [ -</span>
			&nbsp;<span class="label label-warning"> ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--tmp-dir">--tmp-dir </a>
			
		</h3>
		<p class="args">
			<b>Temp directory to use.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">GATKPath</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--use-jdk-deflater">--use-jdk-deflater </a>
			 / <small>-jdk-deflater</small>
		</h3>
		<p class="args">
			<b>Whether to use the JdkDeflater (as opposed to IntelDeflater)</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--use-jdk-inflater">--use-jdk-inflater </a>
			 / <small>-jdk-inflater</small>
		</h3>
		<p class="args">
			<b>Whether to use the JdkInflater (as opposed to IntelInflater)</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--verbosity">--verbosity </a>
			 / <small>-verbosity</small>
		</h3>
		<p class="args">
			<b>Control verbosity of logging.</b><br />
			
		</p>
			<p>
				The --verbosity argument is an enumerated type (LogLevel), which can have one of the following values:
			<dl class="enum">
					<dt class="enum">ERROR</dt>
					<dd class="enum"></dd>
					<dt class="enum">WARNING</dt>
					<dd class="enum"></dd>
					<dt class="enum">INFO</dt>
					<dd class="enum"></dd>
					<dt class="enum">DEBUG</dt>
					<dd class="enum"></dd>
			</dl>
			</p>
		<p>
		<span class="label label-info ">LogLevel</span>
			&nbsp;<span class="label">INFO</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--version">--version </a>
			
		</h3>
		<p class="args">
			<b>display the version number for this tool</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>

        <hr>
        <p><a href='#top'><i class='fa fa-chevron-up'></i> Return to top</a></p>
        <hr>
        <p class="see-also">See also 
        	<a href="https://software.broadinstitute.org/gatk/documentation/">General Documentation</a> |
        	<a class="hide_me_php" href="index.html">Tool Docs Index</a> <a class="hide_me_html" href="index">Tool Documentation Index</a> |
        	<a href="https://gatkforums.broadinstitute.org/gatk/">Support Forum</a>
        </p>

        <p class="version">GATK version 4.3.0.0 built at Wed, 12 Oct 2022 21:04:44 -0400.
        </p>

</div></div>

	<?php printFooter($module); ?>