<?php

	include '../../../../common/include/common.php';
    include_once '../../../config.php';
    $module = modules::GATK;
    $name = docSN::toolDocs;

    printHeader($module, $name, topSN::guide);

    $selected_major_version = 4;
    $dirs = lookupVersionDirs($selected_major_version);
?>

<link type='text/css' rel='stylesheet' href='gatkDoc.css'>

<div class="row hide_me_html" style="margin: 14px 3px 14px 3px;" id="tippy-top">

	<i class='<?php print ico::toolDocsIcon; ?>'></i> <em>Showing tool doc from version 4.3.0.0 | The latest version is <?php print $latest_version; ?></em>
	<div class="btn-group pull-right">
		<a class="btn btn-warning dropdown-toggle" data-toggle="dropdown" href="#">
			4.3.0.0
			<span class="caret"></span>
		</a>
		<ul class="dropdown-menu">
			<?php foreach($dirs as $dir) { ?>
			<li class="hide_me_html"><a tabindex='-1' href='../<?=$dir?>' ><?=$dir?></a></li>
			<?php } ?>
		</ul>
	</div>
	<hr />

</div>


<div class='row' id="top">



	<?php $group = 'Variant Filtering'; ?>

	<section class="col-md-4">
		<aside class="well">
			<a class="hide_me_php" href="index.html"><h4><i class='fa fa-chevron-left'></i> Back to Tool Documentation Index</h4></a>
			<a class="hide_me_html" href=".."><h4><i class='fa fa-chevron-left'></i> Back to Tool Documentation Index</h4></a>
		</aside>
		<aside class="well hide_me_html">
			<h3>Browse tools by category</h3>

        <style>
            #sidenav .accordion-body a {
                color : gray;
            }

            .accordion-body li {
                list-style : none;
            }
        </style>

        <ul class="nav nav-pills nav-stacked" id="sidenav">
        		<hr>
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#CopyNumberVariantDiscovery">Copy Number Variant Discovery</a>
							<div id="CopyNumberVariantDiscovery"
								<?php echo ($group == 'Copy Number Variant Discovery')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="org_broadinstitute_hellbender_tools_copynumber_AnnotateIntervals.html">AnnotateIntervals</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_copynumber_CallCopyRatioSegments.html">CallCopyRatioSegments</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_copynumber_CreateReadCountPanelOfNormals.html">CreateReadCountPanelOfNormals</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_copynumber_DenoiseReadCounts.html">DenoiseReadCounts</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_copynumber_DetermineGermlineContigPloidy.html">DetermineGermlineContigPloidy</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_copynumber_FilterIntervals.html">FilterIntervals</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_copynumber_GermlineCNVCaller.html">GermlineCNVCaller</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_copynumber_ModelSegments.html">ModelSegments</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_copynumber_plotting_PlotDenoisedCopyRatios.html">PlotDenoisedCopyRatios</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_copynumber_plotting_PlotModeledSegments.html">PlotModeledSegments</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_copynumber_PostprocessGermlineCNVCalls.html">PostprocessGermlineCNVCalls</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#CoverageAnalysis">Coverage Analysis</a>
							<div id="CoverageAnalysis"
								<?php echo ($group == 'Coverage Analysis')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_rnaseq_ASEReadCounter.html">ASEReadCounter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_AnalyzeSaturationMutagenesis.html">AnalyzeSaturationMutagenesis</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_copynumber_CollectAllelicCounts.html">CollectAllelicCounts</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_copynumber_CollectReadCounts.html">CollectReadCounts</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_CountBases.html">CountBases</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_pipelines_CountBasesSpark.html">CountBasesSpark</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_CountReads.html">CountReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_pipelines_CountReadsSpark.html">CountReadsSpark</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_coverage_DepthOfCoverage.html">DepthOfCoverage</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_rnaseq_GeneExpressionEvaluation.html">GeneExpressionEvaluation</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_mutect_GetNormalArtifactData.html">GetNormalArtifactData</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_contamination_GetPileupSummaries.html">GetPileupSummaries</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_LocalAssembler.html">LocalAssembler</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_qc_Pileup.html">Pileup</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_PileupSpark.html">PileupSpark</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#DiagnosticsandQualityControl">Diagnostics and Quality Control</a>
							<div id="DiagnosticsandQualityControl"
								<?php echo ($group == 'Diagnostics and Quality Control')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_util_AccumulateQualityYieldMetrics.html">AccumulateQualityYieldMetrics (Picard)</a>
											</li>
											<li>
												<a href="picard_vcf_AccumulateVariantCallingMetrics.html">AccumulateVariantCallingMetrics (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_bqsr_AnalyzeCovariates.html">AnalyzeCovariates</a>
											</li>
											<li>
												<a href="picard_sam_BamIndexStats.html">BamIndexStats (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_sv_evidence_CalcMetadataSpark.html">CalcMetadataSpark</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_contamination_CalculateContamination.html">CalculateContamination</a>
											</li>
											<li>
												<a href="picard_fingerprint_CalculateFingerprintMetrics.html">CalculateFingerprintMetrics (Picard)</a>
											</li>
											<li>
												<a href="picard_sam_CalculateReadGroupChecksum.html">CalculateReadGroupChecksum (Picard)</a>
											</li>
											<li>
												<a href="picard_fingerprint_CheckFingerprint.html">CheckFingerprint (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_qc_CheckPileup.html">CheckPileup</a>
											</li>
											<li>
												<a href="picard_sam_CheckTerminatorBlock.html">CheckTerminatorBlock (Picard)</a>
											</li>
											<li>
												<a href="picard_fingerprint_ClusterCrosscheckMetrics.html">ClusterCrosscheckMetrics (Picard)</a>
											</li>
											<li>
												<a href="picard_analysis_CollectAlignmentSummaryMetrics.html">CollectAlignmentSummaryMetrics (Picard)</a>
											</li>
											<li>
												<a href="picard_arrays_CollectArraysVariantCallingMetrics.html">CollectArraysVariantCallingMetrics (Picard)</a>
											</li>
											<li>
												<a href="picard_analysis_CollectBaseDistributionByCycle.html">CollectBaseDistributionByCycle (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_pipelines_metrics_CollectBaseDistributionByCycleSpark.html">CollectBaseDistributionByCycleSpark</a>
											</li>
											<li>
												<a href="picard_analysis_CollectGcBiasMetrics.html">CollectGcBiasMetrics (Picard)</a>
											</li>
											<li>
												<a href="picard_illumina_quality_CollectHiSeqXPfFailMetrics.html">CollectHiSeqXPfFailMetrics (Picard)</a>
											</li>
											<li>
												<a href="picard_analysis_directed_CollectHsMetrics.html">CollectHsMetrics (Picard)</a>
											</li>
											<li>
												<a href="picard_analysis_replicates_CollectIndependentReplicateMetrics.html">CollectIndependentReplicateMetrics (Picard)</a>
											</li>
											<li>
												<a href="picard_analysis_CollectInsertSizeMetrics.html">CollectInsertSizeMetrics (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_pipelines_metrics_CollectInsertSizeMetricsSpark.html">CollectInsertSizeMetricsSpark</a>
											</li>
											<li>
												<a href="picard_analysis_CollectJumpingLibraryMetrics.html">CollectJumpingLibraryMetrics (Picard)</a>
											</li>
											<li>
												<a href="picard_analysis_CollectMultipleMetrics.html">CollectMultipleMetrics (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_pipelines_metrics_CollectMultipleMetricsSpark.html">CollectMultipleMetricsSpark</a>
											</li>
											<li>
												<a href="picard_analysis_CollectOxoGMetrics.html">CollectOxoGMetrics (Picard)</a>
											</li>
											<li>
												<a href="picard_analysis_CollectQualityYieldMetrics.html">CollectQualityYieldMetrics (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_pipelines_metrics_CollectQualityYieldMetricsSpark.html">CollectQualityYieldMetricsSpark</a>
											</li>
											<li>
												<a href="picard_analysis_CollectRawWgsMetrics.html">CollectRawWgsMetrics (Picard)</a>
											</li>
											<li>
												<a href="picard_analysis_CollectRnaSeqMetrics.html">CollectRnaSeqMetrics (Picard)</a>
											</li>
											<li>
												<a href="picard_analysis_CollectRrbsMetrics.html">CollectRrbsMetrics (Picard)</a>
											</li>
											<li>
												<a href="picard_sam_SamErrorMetric_CollectSamErrorMetrics.html">CollectSamErrorMetrics (Picard)</a>
											</li>
											<li>
												<a href="picard_analysis_artifacts_CollectSequencingArtifactMetrics.html">CollectSequencingArtifactMetrics (Picard)</a>
											</li>
											<li>
												<a href="picard_analysis_directed_CollectTargetedPcrMetrics.html">CollectTargetedPcrMetrics (Picard)</a>
											</li>
											<li>
												<a href="picard_vcf_CollectVariantCallingMetrics.html">CollectVariantCallingMetrics (Picard)</a>
											</li>
											<li>
												<a href="picard_analysis_CollectWgsMetrics.html">CollectWgsMetrics (Picard)</a>
											</li>
											<li>
												<a href="picard_analysis_CollectWgsMetricsWithNonZeroCoverage.html">CollectWgsMetricsWithNonZeroCoverage (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_validation_CompareBaseQualities.html">CompareBaseQualities</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_validation_CompareDuplicatesSpark.html">CompareDuplicatesSpark</a>
											</li>
											<li>
												<a href="picard_analysis_CompareMetrics.html">CompareMetrics (Picard)</a>
											</li>
											<li>
												<a href="picard_sam_CompareSAMs.html">CompareSAMs (Picard)</a>
											</li>
											<li>
												<a href="picard_analysis_artifacts_ConvertSequencingArtifactToOxoG.html">ConvertSequencingArtifactToOxoG (Picard)</a>
											</li>
											<li>
												<a href="picard_fingerprint_CrosscheckFingerprints.html">CrosscheckFingerprints (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_DumpTabixIndex.html">DumpTabixIndex</a>
											</li>
											<li>
												<a href="picard_sam_markduplicates_EstimateLibraryComplexity.html">EstimateLibraryComplexity (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_FlagStat.html">FlagStat</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_pipelines_FlagStatSpark.html">FlagStatSpark</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_GetSampleName.html">GetSampleName</a>
											</li>
											<li>
												<a href="picard_analysis_MeanQualityByCycle.html">MeanQualityByCycle (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_pipelines_metrics_MeanQualityByCycleSpark.html">MeanQualityByCycleSpark</a>
											</li>
											<li>
												<a href="picard_analysis_QualityScoreDistribution.html">QualityScoreDistribution (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_pipelines_metrics_QualityScoreDistributionSpark.html">QualityScoreDistributionSpark</a>
											</li>
											<li>
												<a href="picard_sam_ValidateSamFile.html">ValidateSamFile (Picard)</a>
											</li>
											<li>
												<a href="picard_sam_ViewSam.html">ViewSam (Picard)</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#IntervalsManipulation">Intervals Manipulation</a>
							<div id="IntervalsManipulation"
								<?php echo ($group == 'Intervals Manipulation')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_util_BedToIntervalList.html">BedToIntervalList (Picard)</a>
											</li>
											<li>
												<a href="picard_util_IntervalListToBed.html">IntervalListToBed (Picard)</a>
											</li>
											<li>
												<a href="picard_util_IntervalListTools.html">IntervalListTools (Picard)</a>
											</li>
											<li>
												<a href="picard_util_LiftOverIntervalList.html">LiftOverIntervalList (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_copynumber_PreprocessIntervals.html">PreprocessIntervals</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_SplitIntervals.html">SplitIntervals</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#Metagenomics">Metagenomics</a>
							<div id="Metagenomics"
								<?php echo ($group == 'Metagenomics')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_pathseq_PathSeqBuildKmers.html">PathSeqBuildKmers</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_pathseq_PathSeqBuildReferenceTaxonomy.html">PathSeqBuildReferenceTaxonomy</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_pathseq_PathSeqBwaSpark.html">PathSeqBwaSpark</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_pathseq_PathSeqFilterSpark.html">PathSeqFilterSpark</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_pathseq_PathSeqPipelineSpark.html">PathSeqPipelineSpark</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_pathseq_PathSeqScoreSpark.html">PathSeqScoreSpark</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#Other">Other</a>
							<div id="Other"
								<?php echo ($group == 'Other')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_CreateHadoopBamSplittingIndex.html">CreateHadoopBamSplittingIndex</a>
											</li>
											<li>
												<a href="picard_util_FifoBuffer.html">FifoBuffer (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_bqsr_GatherBQSRReports.html">GatherBQSRReports</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_vqsr_GatherTranches.html">GatherTranches</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_IndexFeatureFile.html">IndexFeatureFile</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_ParallelCopyGCSDirectoryIntoHDFSSpark.html">ParallelCopyGCSDirectoryIntoHDFSSpark</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_ReadAnonymizer.html">ReadAnonymizer</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_variantutils_ReblockGVCF.html">ReblockGVCF</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#ReadDataManipulation">Read Data Manipulation</a>
							<div id="ReadDataManipulation"
								<?php echo ($group == 'Read Data Manipulation')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_sam_AddCommentsToBam.html">AddCommentsToBam (Picard)</a>
											</li>
											<li>
												<a href="picard_sam_AddOATag.html">AddOATag (Picard)</a>
											</li>
											<li>
												<a href="picard_sam_AddOrReplaceReadGroups.html">AddOrReplaceReadGroups (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_bqsr_ApplyBQSR.html">ApplyBQSR</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_ApplyBQSRSpark.html">ApplyBQSRSpark</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_pipelines_BQSRPipelineSpark.html">BQSRPipelineSpark</a>
											</li>
											<li>
												<a href="picard_fastq_BamToBfq.html">BamToBfq (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_bqsr_BaseRecalibrator.html">BaseRecalibrator</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_BaseRecalibratorSpark.html">BaseRecalibratorSpark</a>
											</li>
											<li>
												<a href="picard_sam_BuildBamIndex.html">BuildBamIndex (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_pipelines_BwaAndMarkDuplicatesPipelineSpark.html">BwaAndMarkDuplicatesPipelineSpark</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_bwa_BwaSpark.html">BwaSpark</a>
											</li>
											<li>
												<a href="picard_sam_CleanSam.html">CleanSam (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_ClipReads.html">ClipReads</a>
											</li>
											<li>
												<a href="picard_sam_markduplicates_CollectDuplicateMetrics.html">CollectDuplicateMetrics (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_ConvertHeaderlessHadoopBamShardToBam.html">ConvertHeaderlessHadoopBamShardToBam</a>
											</li>
											<li>
												<a href="picard_sam_DownsampleSam.html">DownsampleSam (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_sv_utils_ExtractOriginalAlignmentRecordsByNameSpark.html">ExtractOriginalAlignmentRecordsByNameSpark</a>
											</li>
											<li>
												<a href="picard_sam_FastqToSam.html">FastqToSam (Picard)</a>
											</li>
											<li>
												<a href="picard_sam_FilterSamReads.html">FilterSamReads (Picard)</a>
											</li>
											<li>
												<a href="picard_sam_FixMateInformation.html">FixMateInformation (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_FixMisencodedBaseQualityReads.html">FixMisencodedBaseQualityReads</a>
											</li>
											<li>
												<a href="picard_sam_GatherBamFiles.html">GatherBamFiles (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_LeftAlignIndels.html">LeftAlignIndels</a>
											</li>
											<li>
												<a href="picard_sam_markduplicates_MarkDuplicates.html">MarkDuplicates (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_transforms_markduplicates_MarkDuplicatesSpark.html">MarkDuplicatesSpark</a>
											</li>
											<li>
												<a href="picard_sam_markduplicates_MarkDuplicatesWithMateCigar.html">MarkDuplicatesWithMateCigar (Picard)</a>
											</li>
											<li>
												<a href="picard_sam_MergeBamAlignment.html">MergeBamAlignment (Picard)</a>
											</li>
											<li>
												<a href="picard_sam_MergeSamFiles.html">MergeSamFiles (Picard)</a>
											</li>
											<li>
												<a href="picard_sam_PositionBasedDownsampleSam.html">PositionBasedDownsampleSam (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_qc_PostProcessReadsForRSEM.html">PostProcessReadsForRSEM</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_PrintDistantMates.html">PrintDistantMates</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_PrintReads.html">PrintReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_PrintReadsHeader.html">PrintReadsHeader</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_pipelines_PrintReadsSpark.html">PrintReadsSpark</a>
											</li>
											<li>
												<a href="picard_sam_ReorderSam.html">ReorderSam (Picard)</a>
											</li>
											<li>
												<a href="picard_sam_ReplaceSamHeader.html">ReplaceSamHeader (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_RevertBaseQualityScores.html">RevertBaseQualityScores</a>
											</li>
											<li>
												<a href="picard_sam_RevertOriginalBaseQualitiesAndAddMateCigar.html">RevertOriginalBaseQualitiesAndAddMateCigar (Picard)</a>
											</li>
											<li>
												<a href="picard_sam_RevertSam.html">RevertSam (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_RevertSamSpark.html">RevertSamSpark</a>
											</li>
											<li>
												<a href="picard_sam_SamFormatConverter.html">SamFormatConverter (Picard)</a>
											</li>
											<li>
												<a href="picard_sam_SamToFastq.html">SamToFastq (Picard)</a>
											</li>
											<li>
												<a href="picard_sam_SetNmAndUqTags.html">SetNmAndUqTags (Picard)</a>
											</li>
											<li>
												<a href="picard_sam_SetNmMdAndUqTags.html">SetNmMdAndUqTags (Picard)</a>
											</li>
											<li>
												<a href="picard_sam_markduplicates_SimpleMarkDuplicatesWithMateCigar.html">SimpleMarkDuplicatesWithMateCigar (Picard)</a>
											</li>
											<li>
												<a href="picard_sam_SortSam.html">SortSam (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_pipelines_SortSamSpark.html">SortSamSpark</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_rnaseq_SplitNCigarReads.html">SplitNCigarReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_SplitReads.html">SplitReads</a>
											</li>
											<li>
												<a href="picard_sam_SplitSamByLibrary.html">SplitSamByLibrary (Picard)</a>
											</li>
											<li>
												<a href="picard_sam_SplitSamByNumberOfReads.html">SplitSamByNumberOfReads (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_qc_TransferReadTags.html">TransferReadTags</a>
											</li>
											<li>
												<a href="picard_sam_markduplicates_UmiAwareMarkDuplicatesWithMateCigar.html">UmiAwareMarkDuplicatesWithMateCigar (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_UnmarkDuplicates.html">UnmarkDuplicates</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#Reference">Reference</a>
							<div id="Reference"
								<?php echo ($group == 'Reference')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_util_BaitDesigner.html">BaitDesigner (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_BwaMemIndexImageCreator.html">BwaMemIndexImageCreator</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_reference_CheckReferenceCompatibility.html">CheckReferenceCompatibility</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_reference_CompareReferences.html">CompareReferences</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_dragstr_ComposeSTRTableFile.html">ComposeSTRTableFile</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_fasta_CountBasesInReference.html">CountBasesInReference</a>
											</li>
											<li>
												<a href="picard_sam_CreateSequenceDictionary.html">CreateSequenceDictionary (Picard)</a>
											</li>
											<li>
												<a href="picard_reference_ExtractSequences.html">ExtractSequences (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_fasta_FastaAlternateReferenceMaker.html">FastaAlternateReferenceMaker</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_fasta_FastaReferenceMaker.html">FastaReferenceMaker</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_sv_evidence_FindBadGenomicKmersSpark.html">FindBadGenomicKmersSpark</a>
											</li>
											<li>
												<a href="picard_reference_NonNFastaSize.html">NonNFastaSize (Picard)</a>
											</li>
											<li>
												<a href="picard_reference_NormalizeFasta.html">NormalizeFasta (Picard)</a>
											</li>
											<li>
												<a href="picard_util_ScatterIntervalsByNs.html">ScatterIntervalsByNs (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_fasta_ShiftFasta.html">ShiftFasta</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#ShortVariantDiscovery">Short Variant Discovery</a>
							<div id="ShortVariantDiscovery"
								<?php echo ($group == 'Short Variant Discovery')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="org_broadinstitute_hellbender_tools_dragstr_CalibrateDragstrModel.html">CalibrateDragstrModel</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_CombineGVCFs.html">CombineGVCFs</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_genomicsdb_GenomicsDBImport.html">GenomicsDBImport</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_GenotypeGVCFs.html">GenotypeGVCFs</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_gnarlyGenotyper_GnarlyGenotyper.html">GnarlyGenotyper</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_variantrecalling_HaplotypeBasedVariantRecaller.html">HaplotypeBasedVariantRecaller</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_haplotypecaller_HaplotypeCaller.html">HaplotypeCaller</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_HaplotypeCallerSpark.html">HaplotypeCallerSpark</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_readorientation_LearnReadOrientationModel.html">LearnReadOrientationModel</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_mutect_Mutect2.html">Mutect2</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_haplotypecaller_RampedHaplotypeCaller.html">RampedHaplotypeCaller</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_pipelines_ReadsPipelineSpark.html">ReadsPipelineSpark</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#StructuralVariantDiscovery">Structural Variant Discovery</a>
							<div id="StructuralVariantDiscovery"
								<?php echo ($group == 'Structural Variant Discovery')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_sv_CollectSVEvidence.html">CollectSVEvidence</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_sv_discovery_inference_CpxVariantReInterpreterSpark.html">CpxVariantReInterpreterSpark</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_sv_discovery_DiscoverVariantsFromContigAlignmentsSAMSpark.html">DiscoverVariantsFromContigAlignmentsSAMSpark</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_sv_evidence_ExtractSVEvidenceSpark.html">ExtractSVEvidenceSpark</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_sv_evidence_FindBreakpointEvidenceSpark.html">FindBreakpointEvidenceSpark</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_sv_JointGermlineCNVSegmentation.html">JointGermlineCNVSegmentation</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_sv_PrintReadCounts.html">PrintReadCounts</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_sv_SVAnnotate.html">SVAnnotate</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_sv_SVCluster.html">SVCluster</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_sv_SiteDepthtoBAF.html">SiteDepthtoBAF</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_sv_StructuralVariationDiscoveryPipelineSpark.html">StructuralVariationDiscoveryPipelineSpark</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_sv_discovery_SvDiscoverFromLocalAssemblyContigAlignmentsSpark.html">SvDiscoverFromLocalAssemblyContigAlignmentsSpark</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantEvaluationandRefinement">Variant Evaluation and Refinement</a>
							<div id="VariantEvaluationandRefinement"
								<?php echo ($group == 'Variant Evaluation and Refinement')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_validation_AnnotateVcfWithBamDepth.html">AnnotateVcfWithBamDepth</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_validation_AnnotateVcfWithExpectedAlleleFraction.html">AnnotateVcfWithExpectedAlleleFraction</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_variantutils_CalculateGenotypePosteriors.html">CalculateGenotypePosteriors</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_validation_CalculateMixingFractions.html">CalculateMixingFractions</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_validation_Concordance.html">Concordance</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_validation_CountFalsePositives.html">CountFalsePositives</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_CountVariants.html">CountVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_pipelines_CountVariantsSpark.html">CountVariantsSpark</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_validation_EvaluateInfoFieldConcordance.html">EvaluateInfoFieldConcordance</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_funcotator_FilterFuncotations.html">FilterFuncotations</a>
											</li>
											<li>
												<a href="picard_vcf_MendelianViolations_FindMendelianViolations.html">FindMendelianViolations (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_funcotator_FuncotateSegments.html">FuncotateSegments</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_funcotator_Funcotator.html">Funcotator</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_funcotator_FuncotatorDataSourceDownloader.html">FuncotatorDataSourceDownloader</a>
											</li>
											<li>
												<a href="picard_vcf_GenotypeConcordance.html">GenotypeConcordance (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_validation_basicshortmutpileup_ValidateBasicSomaticShortMutations.html">ValidateBasicSomaticShortMutations</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_variantutils_ValidateVariants.html">ValidateVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_varianteval_VariantEval.html">VariantEval</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_variantutils_VariantsToTable.html">VariantsToTable</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantFiltering">Variant Filtering</a>
							<div id="VariantFiltering"
								<?php echo ($group == 'Variant Filtering')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_vqsr_ApplyVQSR.html">ApplyVQSR</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_vqsr_CNNScoreVariants.html">CNNScoreVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_vqsr_CNNVariantTrain.html">CNNVariantTrain</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_vqsr_CNNVariantWriteTensors.html">CNNVariantWriteTensors</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_mutect_CreateSomaticPanelOfNormals.html">CreateSomaticPanelOfNormals</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_vqsr_scalable_ExtractVariantAnnotations.html">ExtractVariantAnnotations</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_realignmentfilter_FilterAlignmentArtifacts.html">FilterAlignmentArtifacts</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_mutect_filtering_FilterMutectCalls.html">FilterMutectCalls</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_vqsr_FilterVariantTranches.html">FilterVariantTranches</a>
											</li>
											<li>
												<a href="picard_vcf_filter_FilterVcf.html">FilterVcf (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_vqsr_scalable_ScoreVariantAnnotations.html">ScoreVariantAnnotations</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_vqsr_scalable_TrainVariantAnnotationsModel.html">TrainVariantAnnotationsModel</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_filters_VariantFiltration.html">VariantFiltration</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_vqsr_VariantRecalibrator.html">VariantRecalibrator</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantManipulation">Variant Manipulation</a>
							<div id="VariantManipulation"
								<?php echo ($group == 'Variant Manipulation')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_vcf_FixVcfHeader.html">FixVcfHeader (Picard)</a>
											</li>
											<li>
												<a href="picard_vcf_GatherVcfs.html">GatherVcfs (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_GatherVcfsCloud.html">GatherVcfsCloud</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_variantutils_LeftAlignAndTrimVariants.html">LeftAlignAndTrimVariants</a>
											</li>
											<li>
												<a href="picard_vcf_LiftoverVcf.html">LiftoverVcf (Picard)</a>
											</li>
											<li>
												<a href="picard_vcf_MakeSitesOnlyVcf.html">MakeSitesOnlyVcf (Picard)</a>
											</li>
											<li>
												<a href="picard_vcf_MergeVcfs.html">MergeVcfs (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_spark_pipelines_PrintVariantsSpark.html">PrintVariantsSpark</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_validation_RemoveNearbyIndels.html">RemoveNearbyIndels</a>
											</li>
											<li>
												<a href="picard_vcf_RenameSampleInVcf.html">RenameSampleInVcf (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_variantutils_SelectVariants.html">SelectVariants</a>
											</li>
											<li>
												<a href="picard_vcf_SortVcf.html">SortVcf (Picard)</a>
											</li>
											<li>
												<a href="picard_vcf_SplitVcfs.html">SplitVcfs (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_variantutils_UpdateVCFSequenceDictionary.html">UpdateVCFSequenceDictionary</a>
											</li>
											<li>
												<a href="picard_vcf_UpdateVcfSequenceDictionary.html">UpdateVcfSequenceDictionary (Picard)</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_VariantAnnotator.html">VariantAnnotator</a>
											</li>
											<li>
												<a href="picard_vcf_VcfFormatConverter.html">VcfFormatConverter (Picard)</a>
											</li>
											<li>
												<a href="picard_vcf_VcfToIntervalList.html">VcfToIntervalList (Picard)</a>
											</li>
								</ul>
							</div>
						</li>
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#BaseCalling">Base Calling</a>
							<div id="BaseCalling"
								<?php echo ($group == 'Base Calling')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_illumina_CheckIlluminaDirectory.html">CheckIlluminaDirectory (Picard)</a>
											</li>
											<li>
												<a href="picard_illumina_CollectIlluminaBasecallingMetrics.html">CollectIlluminaBasecallingMetrics (Picard)</a>
											</li>
											<li>
												<a href="picard_illumina_CollectIlluminaLaneMetrics.html">CollectIlluminaLaneMetrics (Picard)</a>
											</li>
											<li>
												<a href="picard_illumina_ExtractIlluminaBarcodes.html">ExtractIlluminaBarcodes (Picard)</a>
											</li>
											<li>
												<a href="picard_illumina_IlluminaBasecallsToFastq.html">IlluminaBasecallsToFastq (Picard)</a>
											</li>
											<li>
												<a href="picard_illumina_IlluminaBasecallsToSam.html">IlluminaBasecallsToSam (Picard)</a>
											</li>
											<li>
												<a href="picard_illumina_MarkIlluminaAdapters.html">MarkIlluminaAdapters (Picard)</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#FlowBasedTools">Flow Based Tools</a>
							<div id="FlowBasedTools"
								<?php echo ($group == 'Flow Based Tools')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_featuremapping_FlowFeatureMapper.html">FlowFeatureMapper</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_groundtruth_GroundTruthReadsBuilder.html">GroundTruthReadsBuilder</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#GenotypingArraysManipulation">Genotyping Arrays Manipulation</a>
							<div id="GenotypingArraysManipulation"
								<?php echo ($group == 'Genotyping Arrays Manipulation')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_arrays_illumina_BpmToNormalizationManifestCsv.html">BpmToNormalizationManifestCsv (Picard)</a>
											</li>
											<li>
												<a href="picard_arrays_CombineGenotypingArrayVcfs.html">CombineGenotypingArrayVcfs (Picard)</a>
											</li>
											<li>
												<a href="picard_arrays_CreateBafRegressMetricsFile.html">CreateBafRegressMetricsFile (Picard)</a>
											</li>
											<li>
												<a href="picard_arrays_CreateVerifyIDIntensityContaminationMetricsFile.html">CreateVerifyIDIntensityContaminationMetricsFile (Picard)</a>
											</li>
											<li>
												<a href="picard_arrays_GtcToVcf.html">GtcToVcf (Picard)</a>
											</li>
											<li>
												<a href="picard_arrays_MergePedIntoVcf.html">MergePedIntoVcf (Picard)</a>
											</li>
											<li>
												<a href="picard_arrays_VcfToAdpc.html">VcfToAdpc (Picard)</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#MethylationSpecificTools">Methylation-Specific Tools</a>
							<div id="MethylationSpecificTools"
								<?php echo ($group == 'Methylation-Specific Tools')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_MethylationTypeCaller.html">MethylationTypeCaller</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#Metrics">Metrics</a>
							<div id="Metrics"
								<?php echo ($group == 'Metrics')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_analysis_AlignmentSummaryMetrics.html">AlignmentSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_arrays_CollectArraysVariantCallingMetrics$ArraysControlCodesSummaryMetrics.html">ArraysControlCodesSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_arrays_CollectArraysVariantCallingMetrics$ArraysVariantCallingDetailMetrics.html">ArraysVariantCallingDetailMetrics</a>
											</li>
											<li>
												<a href="picard_arrays_CollectArraysVariantCallingMetrics$ArraysVariantCallingSummaryMetrics.html">ArraysVariantCallingSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_arrays_BafRegressMetrics.html">BafRegressMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_artifacts_SequencingArtifactMetrics$BaitBiasDetailMetrics.html">BaitBiasDetailMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_artifacts_SequencingArtifactMetrics$BaitBiasSummaryMetrics.html">BaitBiasSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_illumina_BarcodeMetric.html">BarcodeMetric</a>
											</li>
											<li>
												<a href="picard_analysis_BaseDistributionByCycleMetrics.html">BaseDistributionByCycleMetrics</a>
											</li>
											<li>
												<a href="picard_sam_SamErrorMetric_BaseErrorMetric.html">BaseErrorMetric</a>
											</li>
											<li>
												<a href="picard_fingerprint_ClusteredCrosscheckMetric.html">ClusteredCrosscheckMetric</a>
											</li>
											<li>
												<a href="picard_analysis_CollectOxoGMetrics$CpcgMetrics.html">CpcgMetrics</a>
											</li>
											<li>
												<a href="picard_sam_DuplicationMetrics.html">DuplicationMetrics</a>
											</li>
											<li>
												<a href="picard_sam_SamErrorMetric_ErrorMetric.html">ErrorMetric</a>
											</li>
											<li>
												<a href="picard_analysis_artifacts_ErrorSummaryMetrics.html">ErrorSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_fingerprint_FingerprintMetrics.html">FingerprintMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_FingerprintingDetailMetrics.html">FingerprintingDetailMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_FingerprintingSummaryMetrics.html">FingerprintingSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_GcBiasDetailMetrics.html">GcBiasDetailMetrics</a>
											</li>
											<li>
												<a href="picard_metrics_GcBiasMetrics.html">GcBiasMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_GcBiasSummaryMetrics.html">GcBiasSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_vcf_GenotypeConcordanceContingencyMetrics.html">GenotypeConcordanceContingencyMetrics</a>
											</li>
											<li>
												<a href="picard_vcf_GenotypeConcordanceDetailMetrics.html">GenotypeConcordanceDetailMetrics</a>
											</li>
											<li>
												<a href="picard_vcf_GenotypeConcordanceSummaryMetrics.html">GenotypeConcordanceSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_directed_HsMetrics.html">HsMetrics</a>
											</li>
											<li>
												<a href="picard_illumina_IlluminaBasecallingMetrics.html">IlluminaBasecallingMetrics</a>
											</li>
											<li>
												<a href="picard_illumina_IlluminaLaneMetrics.html">IlluminaLaneMetrics</a>
											</li>
											<li>
												<a href="picard_illumina_IlluminaPhasingMetrics.html">IlluminaPhasingMetrics</a>
											</li>
											<li>
												<a href="picard_sam_SamErrorMetric_IndelErrorMetric.html">IndelErrorMetric</a>
											</li>
											<li>
												<a href="picard_analysis_replicates_IndependentReplicateMetric.html">IndependentReplicateMetric</a>
											</li>
											<li>
												<a href="picard_analysis_InsertSizeMetrics.html">InsertSizeMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_JumpingLibraryMetrics.html">JumpingLibraryMetrics</a>
											</li>
											<li>
												<a href="picard_vcf_MendelianViolations_MendelianViolationMetrics.html">MendelianViolationMetrics</a>
											</li>
											<li>
												<a href="picard_sam_SamErrorMetric_OverlappingErrorMetric.html">OverlappingErrorMetric</a>
											</li>
											<li>
												<a href="picard_illumina_quality_CollectHiSeqXPfFailMetrics$PFFailDetailedMetric.html">PFFailDetailedMetric</a>
											</li>
											<li>
												<a href="picard_illumina_quality_CollectHiSeqXPfFailMetrics$PFFailSummaryMetric.html">PFFailSummaryMetric</a>
											</li>
											<li>
												<a href="picard_analysis_artifacts_SequencingArtifactMetrics$PreAdapterDetailMetrics.html">PreAdapterDetailMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_artifacts_SequencingArtifactMetrics$PreAdapterSummaryMetrics.html">PreAdapterSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectQualityYieldMetrics$QualityYieldMetrics.html">QualityYieldMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectRawWgsMetrics$RawWgsMetrics.html">RawWgsMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_RnaSeqMetrics.html">RnaSeqMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_RrbsCpgDetailMetrics.html">RrbsCpgDetailMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_RrbsSummaryMetrics.html">RrbsSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_sam_SamComparisonMetric.html">SamComparisonMetric</a>
											</li>
											<li>
												<a href="picard_analysis_directed_TargetMetrics.html">TargetMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_directed_TargetedPcrMetrics.html">TargetedPcrMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_TheoreticalSensitivityMetrics.html">TheoreticalSensitivityMetrics</a>
											</li>
											<li>
												<a href="picard_sam_markduplicates_UmiMetrics.html">UmiMetrics</a>
											</li>
											<li>
												<a href="picard_vcf_CollectVariantCallingMetrics$VariantCallingDetailMetrics.html">VariantCallingDetailMetrics</a>
											</li>
											<li>
												<a href="picard_vcf_CollectVariantCallingMetrics$VariantCallingSummaryMetrics.html">VariantCallingSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_arrays_VerifyIDIntensityContaminationMetrics.html">VerifyIDIntensityContaminationMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_WgsMetrics.html">WgsMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectWgsMetricsWithNonZeroCoverage$WgsMetricsWithNonZeroCoverage.html">WgsMetricsWithNonZeroCoverage</a>
											</li>
								</ul>
							</div>
						</li>
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#FlowAnnotations">Flow Annotations</a>
							<div id="FlowAnnotations"
								<?php echo ($group == 'Flow Annotations')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_flow_CycleSkipStatus.html">CycleSkipStatus</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_flow_GcContent.html">GcContent</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_HaplotypeFilteringAnnotation.html">HaplotypeFilteringAnnotation</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_flow_HmerIndelLength.html">HmerIndelLength</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_flow_HmerIndelNuc.html">HmerIndelNuc</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_flow_HmerMotifs.html">HmerMotifs</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_flow_IndelClassify.html">IndelClassify</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_flow_IndelLength.html">IndelLength</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_flow_VariantType.html">VariantType</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#ReadFilters">Read Filters</a>
							<div id="ReadFilters"
								<?php echo ($group == 'Read Filters')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_AlignmentAgreesWithHeaderReadFilter.html">AlignmentAgreesWithHeaderReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$AllowAllReadsReadFilter.html">AllowAllReadsReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_AmbiguousBaseReadFilter.html">AmbiguousBaseReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$CigarContainsNoNOperator.html">CigarContainsNoNOperator</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ExcessiveEndClippedReadFilter.html">ExcessiveEndClippedReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$FirstOfPairReadFilter.html">FirstOfPairReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_flow_FlowBasedTPAttributeSymetricReadFilter.html">FlowBasedTPAttributeSymetricReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_flow_FlowBasedTPAttributeValidReadFilter.html">FlowBasedTPAttributeValidReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_FragmentLengthReadFilter.html">FragmentLengthReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$GoodCigarReadFilter.html">GoodCigarReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$HasReadGroupReadFilter.html">HasReadGroupReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_flow_HmerQualitySymetricReadFilter.html">HmerQualitySymetricReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_IntervalOverlapReadFilter.html">IntervalOverlapReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_JexlExpressionReadTagValueFilter.html">JexlExpressionReadTagValueFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_LibraryReadFilter.html">LibraryReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$MappedReadFilter.html">MappedReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$MappingQualityAvailableReadFilter.html">MappingQualityAvailableReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$MappingQualityNotZeroReadFilter.html">MappingQualityNotZeroReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_MappingQualityReadFilter.html">MappingQualityReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$MatchingBasesAndQualsReadFilter.html">MatchingBasesAndQualsReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$MateDifferentStrandReadFilter.html">MateDifferentStrandReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_MateDistantReadFilter.html">MateDistantReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$MateOnSameContigOrNoMappedMateReadFilter.html">MateOnSameContigOrNoMappedMateReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$MateUnmappedAndUnmappedReadFilter.html">MateUnmappedAndUnmappedReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_MetricsReadFilter.html">MetricsReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$NonChimericOriginalAlignmentReadFilter.html">NonChimericOriginalAlignmentReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$NonZeroFragmentLengthReadFilter.html">NonZeroFragmentLengthReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$NonZeroReferenceLengthAlignmentReadFilter.html">NonZeroReferenceLengthAlignmentReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$NotDuplicateReadFilter.html">NotDuplicateReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$NotProperlyPairedReadFilter.html">NotProperlyPairedReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$NotSecondaryAlignmentReadFilter.html">NotSecondaryAlignmentReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$NotSupplementaryAlignmentReadFilter.html">NotSupplementaryAlignmentReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_OverclippedReadFilter.html">OverclippedReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$PairedReadFilter.html">PairedReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$PassesVendorQualityCheckReadFilter.html">PassesVendorQualityCheckReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_PlatformReadFilter.html">PlatformReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_PlatformUnitReadFilter.html">PlatformUnitReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$PrimaryLineReadFilter.html">PrimaryLineReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$ProperlyPairedReadFilter.html">ProperlyPairedReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadGroupBlackListReadFilter.html">ReadGroupBlackListReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_flow_ReadGroupHasFlowOrderReadFilter.html">ReadGroupHasFlowOrderReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadGroupReadFilter.html">ReadGroupReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$ReadLengthEqualsCigarLengthReadFilter.html">ReadLengthEqualsCigarLengthReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadLengthReadFilter.html">ReadLengthReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadNameReadFilter.html">ReadNameReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadStrandFilter.html">ReadStrandFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadTagValueFilter.html">ReadTagValueFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_SampleReadFilter.html">SampleReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$SecondOfPairReadFilter.html">SecondOfPairReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$SeqIsStoredReadFilter.html">SeqIsStoredReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_SoftClippedReadFilter.html">SoftClippedReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$ValidAlignmentEndReadFilter.html">ValidAlignmentEndReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$ValidAlignmentStartReadFilter.html">ValidAlignmentStartReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_flow_WellformedFlowBasedReadFilter.html">WellformedFlowBasedReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_engine_filters_WellformedReadFilter.html">WellformedReadFilter</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantAnnotations">Variant Annotations</a>
							<div id="VariantAnnotations"
								<?php echo ($group == 'Variant Annotations')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_allelespecific_AS_BaseQualityRankSumTest.html">AS_BaseQualityRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_allelespecific_AS_FisherStrand.html">AS_FisherStrand</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_allelespecific_AS_InbreedingCoeff.html">AS_InbreedingCoeff</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_allelespecific_AS_MappingQualityRankSumTest.html">AS_MappingQualityRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_allelespecific_AS_QualByDepth.html">AS_QualByDepth</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_allelespecific_AS_RMSMappingQuality.html">AS_RMSMappingQuality</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_allelespecific_AS_ReadPosRankSumTest.html">AS_ReadPosRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_allelespecific_AS_StrandOddsRatio.html">AS_StrandOddsRatio</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_AlleleFraction.html">AlleleFraction</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_AllelePseudoDepth.html">AllelePseudoDepth</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_AssemblyComplexity.html">AssemblyComplexity</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_BaseQuality.html">BaseQuality</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_BaseQualityRankSumTest.html">BaseQualityRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_ChromosomeCounts.html">ChromosomeCounts</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_ClippingRankSumTest.html">ClippingRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_CountNs.html">CountNs</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_Coverage.html">Coverage</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_DepthPerAlleleBySample.html">DepthPerAlleleBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_DepthPerSampleHC.html">DepthPerSampleHC</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_ExcessHet.html">ExcessHet</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_FeaturizedReadSets.html">FeaturizedReadSets</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_FisherStrand.html">FisherStrand</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_FragmentDepthPerAlleleBySample.html">FragmentDepthPerAlleleBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_FragmentLength.html">FragmentLength</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_GenotypeSummaries.html">GenotypeSummaries</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_InbreedingCoeff.html">InbreedingCoeff</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_LikelihoodRankSumTest.html">LikelihoodRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_MappingQuality.html">MappingQuality</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_MappingQualityRankSumTest.html">MappingQualityRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_MappingQualityZero.html">MappingQualityZero</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_OrientationBiasReadCounts.html">OrientationBiasReadCounts</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_OriginalAlignment.html">OriginalAlignment</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_PossibleDeNovo.html">PossibleDeNovo</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_QualByDepth.html">QualByDepth</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_RMSMappingQuality.html">RMSMappingQuality</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_RawGtCount.html">RawGtCount</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_ReadPosRankSumTest.html">ReadPosRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_ReadPosition.html">ReadPosition</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_ReferenceBases.html">ReferenceBases</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_SampleList.html">SampleList</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_StrandBiasBySample.html">StrandBiasBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_StrandOddsRatio.html">StrandOddsRatio</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_TandemRepeat.html">TandemRepeat</a>
											</li>
											<li>
												<a href="org_broadinstitute_hellbender_tools_walkers_annotator_UniqueAltReadCount.html">UniqueAltReadCount</a>
											</li>
								</ul>
							</div>
						</li>
        </ul>
		</aside>

	</section>

	<div class="col-md-8">

			<h1>**BETA** ExtractVariantAnnotations</h1>

			<p>Extracts site-level variant annotations, labels, and other metadata from a VCF file to HDF5 files</p>

			<h3>Category
				<small> Variant Filtering</small>
			</h3>
		<hr>
		<h2>Overview</h2>
		Extracts site-level variant annotations, labels, and other metadata from a VCF file to HDF5 files.

 <p>
     This tool is intended to be used as the first step in a variant-filtering workflow that supersedes the
     VariantRecalibrator workflow. This tool extracts site-level annotations, labels, and other relevant metadata
     from variant sites (or alleles, in allele-specific mode) that are or are not present in specified labeled
     resource VCFs (e.g., training or calibration VCFs). Input sites that are present in the resources are considered
     labeled; each site can have multiple labels if it is present in multiple resources. Other input sites that are
     not present in any resources are considered unlabeled and can be randomly sampled using reservoir sampling;
     extraction of these is optional. The outputs of the tool are HDF5 files containing the extracted data for
     labeled and (optional) unlabeled variant sets, as well as a sites-only indexed VCF containing the labeled variants.
 </p>
 
 <p>
     The extracted sets can be provided as input to the TrainVariantAnnotationsModel tool
     to produce an annotation-based model for scoring variant calls. This model can in turn be provided
     along with a VCF file to the ScoreVariantAnnotations tool, which assigns a score to each call
     (with a lower score indicating that a call is more likely to be an artifact and should perhaps be filtered).
     Each score can also be converted to a corresponding sensitivity with respect to a calibration set, if the latter is available.
 </p>

 <p>
     Note that annotations and metadata are collected in memory during traversal until they are written to HDF5 files
     upon completion of the traversal. Memory requirements thus roughly scale linearly with both the number of sites
     extracted and the number of annotations.
 </p>

 <p>
     Note that HDF5 files may be viewed using <a href="https://support.hdfgroup.org/products/java/hdfview/">hdfview</a>
     or loaded in Python using <a href="http://www.pytables.org/">PyTables</a> or <a href="http://www.h5py.org/">h5py</a>.
 </p>
 
 <h3>Inputs</h3>

 <ul>
     <li>
         Input VCF file. Site-level annotations will be extracted from the contained variants (or alleles, 
         if the USE_ALLELE_SPECIFIC_ANNOTATIONS_LONG_NAME argument is specified).
     </li>
     <li>
         Annotations to extract.
     </li>
     <li>
         Variant types (i.e., SNP and/or INDEL) to extract. Logic for determining variant type was retained from
         VariantRecalibrator; see VariantType. Extracting SNPs and INDELs separately in two runs of
         this tool can be useful if one wishes to extract different sets of annotations for each variant type,
         for example.
     </li>
     <li>
         (Optional) Resource VCF file(s). Each resource should be tagged with a label, which will be assigned to
         extracted sites that are present in the resource. In typical use, the LabeledVariantAnnotationsData#TRAINING_LABEL
         and LabeledVariantAnnotationsData#CALIBRATION_LABEL labels should be used to tag at least one resource
         apiece. The resulting sets of sites will be used for model training and conversion of scores to
         calibration-set sensitivity, respectively; the trustworthiness of the respective resources should be
         taken into account accordingly. The LabeledVariantAnnotationsData#SNP_LABEL label is
         reserved by the tool, as it is used to label sites determined to be SNPs, and thus it cannot be used to tag
         provided resources.
     </li>
     <li>
         (Optional) Maximum number of unlabeled variants (or alleles) to randomly sample with reservoir sampling.
         If nonzero, annotations will also be extracted from unlabeled sites (i.e., those that are not present
         in the labeled resources).
     </li>
     <li>
         Output prefix.
         This is used as the basename for output files.
     </li>
 </ul>

 <h3>Outputs</h3>

 <ul>
     <li>
         (Optional) Labeled-annotations HDF5 file (.annot.hdf5). Annotation data and metadata for those sites that
         are present in labeled resources are stored in the following HDF5 directory structure:

         <p>
           |--- alleles<br>
           |    |--- alt<br>
           |    |--- ref<br>
           |--- annotations<br>
           |    |--- chunk_0<br>
           |    |--- ...<br>
           |    |--- chunk_{num_chunks - 1}<br>
           |    |--- names<br>
           |    |--- num_chunks<br>
           |    |--- num_columns<br>
           |    |--- num_rows<br>
           |--- intervals<br>
           |    |--- indexed_contig_names<br>
           |    |--- transposed_index_start_end<br>
           |--- labels<br>
           |    |--- snp<br>
           |    |--- ... (e.g., training, calibration, etc.)<br>
           |    |--- ...<br>
         </p>

         <p>
             Here, each chunk is a double matrix, with dimensions given by (number of sites in the chunk) x (number of annotations).
             See the methods HDF5Utils#writeChunkedDoubleMatrix and HDF5Utils#writeIntervals for additional details.
             If USE_ALLELE_SPECIFIC_ANNOTATIONS_LONG_NAME is specified, each record corresponds to an individual allele;
             otherwise, each record corresponds to a variant site, which may contain multiple alleles.
             Storage of alleles can be omitted using the OMIT_ALLELES_IN_HDF5_LONG_NAME argument, which will reduce
             the size of the file. This file will only be produced if resources are provided and the number of extracted
             labeled sites is nonzero.
         </p>

     </li>
     <li>
         Labeled sites-only VCF file and index. The VCF will not be gzipped if the DO_NOT_GZIP_VCF_OUTPUT_LONG_NAME
         argument is set to true. The VCF can be provided as a resource in subsequent runs of
         ScoreVariantAnnotations and used to indicate labeled sites that were extracted.
         This can be useful if the StandardArgumentDefinitions#INTERVALS_LONG_NAME argument was used to
         subset sites in training or calibration resources for extraction; this may occur when setting up
         training/validation/test splits, for example. Note that records for the random sample of unlabeled sites are
         currently not included in the VCF.
     </li>
     <li>
         (Optional) Unlabeled-annotations HDF5 file. This will have the same directory structure as in the
         labeled-annotations HDF5 file. However, note that records are currently written in the order they
         appear in the downsampling reservoir after random sampling, and hence, are not in genomic order.
         This file will only be produced if a nonzero value of the MAXIMUM_NUMBER_OF_UNLABELED_VARIANTS_LONG_NAME
         argument is provided.
     </li>
 </ul>

 <h3>Usage examples</h3>

 <p>
     Extract annotations from training/calibration SNP/INDEL sites, producing the outputs
     1) extract.annot.hdf5, 2) extract.vcf.gz, and 3) extract.vcf.gz.tbi.
     The HDF5 file can then be provided to TrainVariantAnnotationsModel
     to train a model using a positive-only approach. Note that the MODE_LONG_NAME arguments are made
     explicit here, although both SNP and INDEL modes are selected by default.

 <pre>
     gatk ExtractVariantAnnotations \
          -V input.vcf \
          -A annotation_1 \
          ...
          -A annotation_N \
          --mode SNP \
          --resource:snp-training,training=true snp-training.vcf \
          --resource:snp-calibration,calibration=true snp-calibration.vcf \
          --mode INDEL \
          --resource:indel-training,training=true indel-training.vcf \
          --resource:indel-calibration,calibration=true indel-calibration.vcf \
          -O extract
 </pre>
 </p>

 <p>
     Extract annotations from both training/calibration SNP/INDEL sites and a random sample of
     1000000 unlabeled (i.e., non-training/calibration) sites, producing the outputs
     1) extract.annot.hdf5, 2) extract.unlabeled.annot.hdf5, 3) extract.vcf.gz,
     and 4) extract.vcf.gz.tbi. The HDF5 files can then be provided to TrainVariantAnnotationsModel
     to train a model using a positive-negative approach (similar to that used in VariantRecalibrator).
     Note that the MODE_LONG_NAME arguments are made explicit here, although both SNP and INDEL modes are
     selected by default.

 <pre>
     gatk ExtractVariantAnnotations \
          -V input.vcf \
          -A annotation_1 \
          ...
          -A annotation_N \
          --mode SNP \
          --resource:snp-training,training=true snp-training.vcf \
          --resource:snp-calibration,calibration=true snp-calibration.vcf \
          --mode INDEL \
          --resource:indel-training,training=true indel-training.vcf \
          --resource:indel-calibration,calibration=true indel-calibration.vcf \
          --maximum-number-of-unlableled-variants 1000000
          -O extract
 </pre>
 </p>

 <p>
     In the (atypical) event that resource VCFs are unavailable, one can still extract annotations from a random sample of
     unlabeled sites, producing the outputs 1) extract.unlabeled.annot.hdf5,
     2) extract.vcf.gz (which will contain no records), and 3) extract.vcf.gz.tbi.
     This random sample cannot be used by TrainVariantAnnotationsModel, but may still be useful for
     exploratory analyses. Note that the MODE_LONG_NAME arguments are made explicit here, although both
     SNP and INDEL modes are selected by default.

 <pre>
     gatk ExtractVariantAnnotations \
          -V input.vcf \
          -A annotation_1 \
          ...
          -A annotation_N \
          --mode SNP \
          --mode INDEL \
          --maximum-number-of-unlableled-variants 1000000
          -O extract
 </pre>
 </p>

 DEVELOPER NOTE: See documentation in LabeledVariantAnnotationsWalker.

				<hr>
				<h2>Additional Information</h2>
				<p></p>
				<h3>Read filters</h3>
					<p>This Read Filter is automatically applied to the data by the Engine before processing by ExtractVariantAnnotations.</p>
				<ul>
						<li><a href="org_broadinstitute_hellbender_engine_filters_WellformedReadFilter.html">WellformedReadFilter</a></li>
				</ul>

				<h3>ExtractVariantAnnotations specific arguments</h3>
				<p>This table summarizes the command-line arguments that are specific to this tool. For more details on each argument, see the list further down below the table or click on an argument name to jump directly to that entry in the list.</p>
				<table class="table table-striped table-bordered table-condensed cozy">
					<thead>
					<tr>
						<th>Argument name(s)</th>
						<th>Default value</th>
						<th>Summary</th>
					</tr>
					</thead>
					<tbody>
			<tr>
				<th colspan="4" id="row-divider">Required Arguments</th>
			</tr>
				<tr>
					<td><a href="#--annotation">--annotation</a><br />
								&nbsp;<em>-A</em>
					</td>
					<!--<td>List[String]</td> -->
						<td></td>
					<td>Names of the annotations to extract. Note that a requested annotation may in fact not be present at any extraction site; NaN missing values will be generated for such annotations.</td>
				</tr>
				<tr>
					<td><a href="#--output">--output</a><br />
								&nbsp;<em>-O</em>
					</td>
					<!--<td>String</td> -->
						<td></td>
					<td>Prefix for output filenames.</td>
				</tr>
				<tr>
					<td><a href="#--variant">--variant</a><br />
								&nbsp;<em>-V</em>
					</td>
					<!--<td>GATKPath</td> -->
						<td></td>
					<td>A VCF file containing variants</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Optional Tool Arguments</th>
			</tr>
				<tr>
					<td><a href="#--arguments_file">--arguments_file</a><br />
					</td>
					<!--<td>List[File]</td> -->
						<td></td>
					<td>read one or more arguments files and add them to the command line</td>
				</tr>
				<tr>
					<td><a href="#--cloud-index-prefetch-buffer">--cloud-index-prefetch-buffer</a><br />
								&nbsp;<em>-CIPB</em>
					</td>
					<!--<td>int</td> -->
						<td>-1</td>
					<td>Size of the cloud-only prefetch buffer (in MB; 0 to disable). Defaults to cloudPrefetchBuffer if unset.</td>
				</tr>
				<tr>
					<td><a href="#--cloud-prefetch-buffer">--cloud-prefetch-buffer</a><br />
								&nbsp;<em>-CPB</em>
					</td>
					<!--<td>int</td> -->
						<td>40</td>
					<td>Size of the cloud-only prefetch buffer (in MB; 0 to disable).</td>
				</tr>
				<tr>
					<td><a href="#--disable-bam-index-caching">--disable-bam-index-caching</a><br />
								&nbsp;<em>-DBIC</em>
					</td>
					<!--<td>boolean</td> -->
						<td>false</td>
					<td>If true, don't cache bam indexes, this will reduce memory requirements but may harm performance if many intervals are specified.  Caching is automatically disabled if there are no intervals specified.</td>
				</tr>
				<tr>
					<td><a href="#--disable-sequence-dictionary-validation">--disable-sequence-dictionary-validation</a><br />
					</td>
					<!--<td>boolean</td> -->
						<td>false</td>
					<td>If specified, do not check the sequence dictionaries from our inputs for compatibility. Use at your own risk!</td>
				</tr>
				<tr>
					<td><a href="#--do-not-gzip-vcf-output">--do-not-gzip-vcf-output</a><br />
					</td>
					<!--<td>boolean</td> -->
						<td>false</td>
					<td>If true, VCF output will not be compressed.</td>
				</tr>
				<tr>
					<td><a href="#--do-not-trust-all-polymorphic">--do-not-trust-all-polymorphic</a><br />
					</td>
					<!--<td>boolean</td> -->
						<td>false</td>
					<td>If true, do not trust that unfiltered records in the resources contain only polymorphic sites. This may increase runtime if the resources are not sites-only VCFs.</td>
				</tr>
				<tr>
					<td><a href="#--gcs-max-retries">--gcs-max-retries</a><br />
								&nbsp;<em>-gcs-retries</em>
					</td>
					<!--<td>int</td> -->
						<td>20</td>
					<td>If the GCS bucket channel errors out, how many times it will attempt to re-initiate the connection</td>
				</tr>
				<tr>
					<td><a href="#--gcs-project-for-requester-pays">--gcs-project-for-requester-pays</a><br />
					</td>
					<!--<td>String</td> -->
						<td></td>
					<td>Project to bill when accessing "requester pays" buckets. If unset, these buckets cannot be accessed.  User must have storage.buckets.get permission on the bucket being accessed.</td>
				</tr>
				<tr>
					<td><a href="#--help">--help</a><br />
								&nbsp;<em>-h</em>
					</td>
					<!--<td>boolean</td> -->
						<td>false</td>
					<td>display the help message</td>
				</tr>
				<tr>
					<td><a href="#--ignore-all-filters">--ignore-all-filters</a><br />
					</td>
					<!--<td>boolean</td> -->
						<td>false</td>
					<td>If true, ignore all filters in the input VCF.</td>
				</tr>
				<tr>
					<td><a href="#--ignore-filter">--ignore-filter</a><br />
					</td>
					<!--<td>List[String]</td> -->
						<td></td>
					<td>Ignore the specified filter(s) in the input VCF.</td>
				</tr>
				<tr>
					<td><a href="#--interval-merging-rule">--interval-merging-rule</a><br />
								&nbsp;<em>-imr</em>
					</td>
					<!--<td>IntervalMergingRule</td> -->
						<td>ALL</td>
					<td>Interval merging rule for abutting intervals</td>
				</tr>
				<tr>
					<td><a href="#--intervals">--intervals</a><br />
								&nbsp;<em>-L</em>
					</td>
					<!--<td>List[String]</td> -->
						<td></td>
					<td>One or more genomic intervals over which to operate</td>
				</tr>
				<tr>
					<td><a href="#--maximum-number-of-unlabeled-variants">--maximum-number-of-unlabeled-variants</a><br />
					</td>
					<!--<td>int</td> -->
						<td>0</td>
					<td>Maximum number of unlabeled variants to extract. If greater than zero, reservoir sampling will be used to randomly sample this number of sites from input sites that are not present in the specified resources. Choice of this number should be guided by considerations for training the negative model in TrainVariantAnnotationsModel; users may wish to choose a number that is comparable to the expected size of the labeled training set or that is compatible with available memory resources.</td>
				</tr>
				<tr>
					<td><a href="#--mode">--mode</a><br />
					</td>
					<!--<td>List[VariantType]</td> -->
						<td>[SNP, INDEL]</td>
					<td>Variant types to extract.</td>
				</tr>
				<tr>
					<td><a href="#--omit-alleles-in-hdf5">--omit-alleles-in-hdf5</a><br />
					</td>
					<!--<td>boolean</td> -->
						<td>false</td>
					<td>If true, omit alleles in output HDF5 files in order to decrease file sizes.</td>
				</tr>
				<tr>
					<td><a href="#--reference">--reference</a><br />
								&nbsp;<em>-R</em>
					</td>
					<!--<td>GATKPath</td> -->
						<td></td>
					<td>Reference sequence</td>
				</tr>
				<tr>
					<td><a href="#--reservoir-sampling-random-seed">--reservoir-sampling-random-seed</a><br />
					</td>
					<!--<td>int</td> -->
						<td>0</td>
					<td>Random seed to use for reservoir sampling of unlabeled variants.</td>
				</tr>
				<tr>
					<td><a href="#--resource">--resource</a><br />
					</td>
					<!--<td>List[FeatureInput[VariantContext]]</td> -->
						<td></td>
					<td>Resource VCFs used to label extracted variants.</td>
				</tr>
				<tr>
					<td><a href="#--resource-matching-strategy">--resource-matching-strategy</a><br />
					</td>
					<!--<td>ResourceMatchingStrategy</td> -->
						<td>START_POSITION</td>
					<td>The strategy to use for determining whether an input variant is present in a resource in non-allele-specific mode (--use-allele-specific-annotations false). START_POSITION: Start positions of input and resource variants must match. START_POSITION_AND_GIVEN_REPRESENTATION: The intersection of the sets of input and resource alleles (in their given representations) must also be non-empty. START_POSITION_AND_MINIMAL_REPRESENTATION: The intersection of the sets of input and resource alleles (after converting alleles to their minimal representations) must also be non-empty. This argument has no effect in allele-specific mode (--use-allele-specific-annotations true), in which the minimal representations of the input and resource alleles must match.</td>
				</tr>
				<tr>
					<td><a href="#--sites-only-vcf-output">--sites-only-vcf-output</a><br />
					</td>
					<!--<td>boolean</td> -->
						<td>false</td>
					<td>If true, don't emit genotype fields when writing vcf file output.</td>
				</tr>
				<tr>
					<td><a href="#--use-allele-specific-annotations">--use-allele-specific-annotations</a><br />
					</td>
					<!--<td>boolean</td> -->
						<td>false</td>
					<td>If true, use the allele-specific versions of the specified annotations.</td>
				</tr>
				<tr>
					<td><a href="#--version">--version</a><br />
					</td>
					<!--<td>boolean</td> -->
						<td>false</td>
					<td>display the version number for this tool</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Optional Common Arguments</th>
			</tr>
				<tr>
					<td><a href="#--add-output-sam-program-record">--add-output-sam-program-record</a><br />
					</td>
					<!--<td>boolean</td> -->
						<td>true</td>
					<td>If true, adds a PG tag to created SAM/BAM/CRAM files.</td>
				</tr>
				<tr>
					<td><a href="#--add-output-vcf-command-line">--add-output-vcf-command-line</a><br />
					</td>
					<!--<td>boolean</td> -->
						<td>true</td>
					<td>If true, adds a command line header line to created VCF files.</td>
				</tr>
				<tr>
					<td><a href="#--create-output-bam-index">--create-output-bam-index</a><br />
								&nbsp;<em>-OBI</em>
					</td>
					<!--<td>boolean</td> -->
						<td>true</td>
					<td>If true, create a BAM/CRAM index when writing a coordinate-sorted BAM/CRAM file.</td>
				</tr>
				<tr>
					<td><a href="#--create-output-bam-md5">--create-output-bam-md5</a><br />
								&nbsp;<em>-OBM</em>
					</td>
					<!--<td>boolean</td> -->
						<td>false</td>
					<td>If true, create a MD5 digest for any BAM/SAM/CRAM file created</td>
				</tr>
				<tr>
					<td><a href="#--create-output-variant-index">--create-output-variant-index</a><br />
								&nbsp;<em>-OVI</em>
					</td>
					<!--<td>boolean</td> -->
						<td>true</td>
					<td>If true, create a VCF index when writing a coordinate-sorted VCF file.</td>
				</tr>
				<tr>
					<td><a href="#--create-output-variant-md5">--create-output-variant-md5</a><br />
								&nbsp;<em>-OVM</em>
					</td>
					<!--<td>boolean</td> -->
						<td>false</td>
					<td>If true, create a a MD5 digest any VCF file created.</td>
				</tr>
				<tr>
					<td><a href="#--disable-read-filter">--disable-read-filter</a><br />
								&nbsp;<em>-DF</em>
					</td>
					<!--<td>List[String]</td> -->
						<td></td>
					<td>Read filters to be disabled before analysis</td>
				</tr>
				<tr>
					<td><a href="#--disable-tool-default-read-filters">--disable-tool-default-read-filters</a><br />
					</td>
					<!--<td>boolean</td> -->
						<td>false</td>
					<td>Disable all tool default read filters (WARNING: many tools will not function correctly without their default read filters on)</td>
				</tr>
				<tr>
					<td><a href="#--exclude-intervals">--exclude-intervals</a><br />
								&nbsp;<em>-XL</em>
					</td>
					<!--<td>List[String]</td> -->
						<td></td>
					<td>One or more genomic intervals to exclude from processing</td>
				</tr>
				<tr>
					<td><a href="#--gatk-config-file">--gatk-config-file</a><br />
					</td>
					<!--<td>String</td> -->
						<td></td>
					<td>A configuration file to use with the GATK.</td>
				</tr>
				<tr>
					<td><a href="#--input">--input</a><br />
								&nbsp;<em>-I</em>
					</td>
					<!--<td>List[GATKPath]</td> -->
						<td></td>
					<td>BAM/SAM/CRAM file containing reads</td>
				</tr>
				<tr>
					<td><a href="#--interval-exclusion-padding">--interval-exclusion-padding</a><br />
								&nbsp;<em>-ixp</em>
					</td>
					<!--<td>int</td> -->
						<td>0</td>
					<td>Amount of padding (in bp) to add to each interval you are excluding.</td>
				</tr>
				<tr>
					<td><a href="#--interval-padding">--interval-padding</a><br />
								&nbsp;<em>-ip</em>
					</td>
					<!--<td>int</td> -->
						<td>0</td>
					<td>Amount of padding (in bp) to add to each interval you are including.</td>
				</tr>
				<tr>
					<td><a href="#--interval-set-rule">--interval-set-rule</a><br />
								&nbsp;<em>-isr</em>
					</td>
					<!--<td>IntervalSetRule</td> -->
						<td>UNION</td>
					<td>Set merging approach to use for combining interval inputs</td>
				</tr>
				<tr>
					<td><a href="#--lenient">--lenient</a><br />
								&nbsp;<em>-LE</em>
					</td>
					<!--<td>boolean</td> -->
						<td>false</td>
					<td>Lenient processing of VCF files</td>
				</tr>
				<tr>
					<td><a href="#--max-variants-per-shard">--max-variants-per-shard</a><br />
					</td>
					<!--<td>int</td> -->
						<td>0</td>
					<td>If non-zero, partitions VCF output into shards, each containing up to the given number of records.</td>
				</tr>
				<tr>
					<td><a href="#--QUIET">--QUIET</a><br />
					</td>
					<!--<td>Boolean</td> -->
						<td>false</td>
					<td>Whether to suppress job-summary info on System.err.</td>
				</tr>
				<tr>
					<td><a href="#--read-filter">--read-filter</a><br />
								&nbsp;<em>-RF</em>
					</td>
					<!--<td>List[String]</td> -->
						<td></td>
					<td>Read filters to be applied before analysis</td>
				</tr>
				<tr>
					<td><a href="#--read-index">--read-index</a><br />
					</td>
					<!--<td>List[GATKPath]</td> -->
						<td></td>
					<td>Indices to use for the read inputs. If specified, an index must be provided for every read input and in the same order as the read inputs. If this argument is not specified, the path to the index for each input will be inferred automatically.</td>
				</tr>
				<tr>
					<td><a href="#--read-validation-stringency">--read-validation-stringency</a><br />
								&nbsp;<em>-VS</em>
					</td>
					<!--<td>ValidationStringency</td> -->
						<td>SILENT</td>
					<td>Validation stringency for all SAM/BAM/CRAM/SRA files read by this program.  The default stringency value SILENT can improve performance when processing a BAM file in which variable-length data (read, qualities, tags) do not otherwise need to be decoded.</td>
				</tr>
				<tr>
					<td><a href="#--seconds-between-progress-updates">--seconds-between-progress-updates</a><br />
					</td>
					<!--<td>double</td> -->
						<td>10.0</td>
					<td>Output traversal statistics every time this many seconds elapse</td>
				</tr>
				<tr>
					<td><a href="#--sequence-dictionary">--sequence-dictionary</a><br />
					</td>
					<!--<td>GATKPath</td> -->
						<td></td>
					<td>Use the given sequence dictionary as the master/canonical sequence dictionary.  Must be a .dict file.</td>
				</tr>
				<tr>
					<td><a href="#--tmp-dir">--tmp-dir</a><br />
					</td>
					<!--<td>GATKPath</td> -->
						<td></td>
					<td>Temp directory to use.</td>
				</tr>
				<tr>
					<td><a href="#--use-jdk-deflater">--use-jdk-deflater</a><br />
								&nbsp;<em>-jdk-deflater</em>
					</td>
					<!--<td>boolean</td> -->
						<td>false</td>
					<td>Whether to use the JdkDeflater (as opposed to IntelDeflater)</td>
				</tr>
				<tr>
					<td><a href="#--use-jdk-inflater">--use-jdk-inflater</a><br />
								&nbsp;<em>-jdk-inflater</em>
					</td>
					<!--<td>boolean</td> -->
						<td>false</td>
					<td>Whether to use the JdkInflater (as opposed to IntelInflater)</td>
				</tr>
				<tr>
					<td><a href="#--verbosity">--verbosity</a><br />
					</td>
					<!--<td>LogLevel</td> -->
						<td>INFO</td>
					<td>Control verbosity of logging.</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Advanced Arguments</th>
			</tr>
				<tr>
					<td><a href="#--showHidden">--showHidden</a><br />
					</td>
					<!--<td>boolean</td> -->
						<td>false</td>
					<td>display hidden arguments</td>
				</tr>
					</tbody>
				</table>

					<h3>Argument details</h3>
					<p>Arguments in this list are specific to this tool. Keep in mind that other arguments are available that are shared with other tools (e.g. command-line GATK arguments); see Inherited arguments above.</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--add-output-sam-program-record">--add-output-sam-program-record </a>
			 / <small>-add-output-sam-program-record</small>
		</h3>
		<p class="args">
			<b>If true, adds a PG tag to created SAM/BAM/CRAM files.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">true</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--add-output-vcf-command-line">--add-output-vcf-command-line </a>
			 / <small>-add-output-vcf-command-line</small>
		</h3>
		<p class="args">
			<b>If true, adds a command line header line to created VCF files.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">true</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--annotation">--annotation </a>
			 / <small>-A</small>
		</h3>
		<p class="args">
			<b>Names of the annotations to extract. Note that a requested annotation may in fact not be present at any extraction site; NaN missing values will be generated for such annotations.</b><br />
			
		</p>
		<p>
				<span class="badge badge-important">R</span>
		<span class="label label-info ">List[String]</span>
			&nbsp;<span class="label">[]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--arguments_file">--arguments_file </a>
			
		</h3>
		<p class="args">
			<b>read one or more arguments files and add them to the command line</b><br />
			
		</p>
		<p>
		<span class="label label-info ">List[File]</span>
			&nbsp;<span class="label">[]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--cloud-index-prefetch-buffer">--cloud-index-prefetch-buffer </a>
			 / <small>-CIPB</small>
		</h3>
		<p class="args">
			<b>Size of the cloud-only prefetch buffer (in MB; 0 to disable). Defaults to cloudPrefetchBuffer if unset.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">int</span>
			&nbsp;<span class="label">-1</span>
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--cloud-prefetch-buffer">--cloud-prefetch-buffer </a>
			 / <small>-CPB</small>
		</h3>
		<p class="args">
			<b>Size of the cloud-only prefetch buffer (in MB; 0 to disable).</b><br />
			
		</p>
		<p>
		<span class="label label-info ">int</span>
			&nbsp;<span class="label">40</span>
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--create-output-bam-index">--create-output-bam-index </a>
			 / <small>-OBI</small>
		</h3>
		<p class="args">
			<b>If true, create a BAM/CRAM index when writing a coordinate-sorted BAM/CRAM file.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">true</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--create-output-bam-md5">--create-output-bam-md5 </a>
			 / <small>-OBM</small>
		</h3>
		<p class="args">
			<b>If true, create a MD5 digest for any BAM/SAM/CRAM file created</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--create-output-variant-index">--create-output-variant-index </a>
			 / <small>-OVI</small>
		</h3>
		<p class="args">
			<b>If true, create a VCF index when writing a coordinate-sorted VCF file.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">true</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--create-output-variant-md5">--create-output-variant-md5 </a>
			 / <small>-OVM</small>
		</h3>
		<p class="args">
			<b>If true, create a a MD5 digest any VCF file created.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--disable-bam-index-caching">--disable-bam-index-caching </a>
			 / <small>-DBIC</small>
		</h3>
		<p class="args">
			<b>If true, don't cache bam indexes, this will reduce memory requirements but may harm performance if many intervals are specified.  Caching is automatically disabled if there are no intervals specified.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--disable-read-filter">--disable-read-filter </a>
			 / <small>-DF</small>
		</h3>
		<p class="args">
			<b>Read filters to be disabled before analysis</b><br />
			
		</p>
		<p>
		<span class="label label-info ">List[String]</span>
			&nbsp;<span class="label">[]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--disable-sequence-dictionary-validation">--disable-sequence-dictionary-validation </a>
			 / <small>-disable-sequence-dictionary-validation</small>
		</h3>
		<p class="args">
			<b>If specified, do not check the sequence dictionaries from our inputs for compatibility. Use at your own risk!</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--disable-tool-default-read-filters">--disable-tool-default-read-filters </a>
			 / <small>-disable-tool-default-read-filters</small>
		</h3>
		<p class="args">
			<b>Disable all tool default read filters (WARNING: many tools will not function correctly without their default read filters on)</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--do-not-gzip-vcf-output">--do-not-gzip-vcf-output </a>
			
		</h3>
		<p class="args">
			<b>If true, VCF output will not be compressed.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--do-not-trust-all-polymorphic">--do-not-trust-all-polymorphic </a>
			
		</h3>
		<p class="args">
			<b>If true, do not trust that unfiltered records in the resources contain only polymorphic sites. This may increase runtime if the resources are not sites-only VCFs.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--exclude-intervals">--exclude-intervals </a>
			 / <small>-XL</small>
		</h3>
		<p class="args">
			<b>One or more genomic intervals to exclude from processing</b><br />
			Use this argument to exclude certain parts of the genome from the analysis (like -L, but the opposite).
 This argument can be specified multiple times. You can use samtools-style intervals either explicitly on the
 command line (e.g. -XL 1 or -XL 1:100-200) or by loading in a file containing a list of intervals
 (e.g. -XL myFile.intervals).
		</p>
		<p>
		<span class="label label-info ">List[String]</span>
			&nbsp;<span class="label">[]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--gatk-config-file">--gatk-config-file </a>
			
		</h3>
		<p class="args">
			<b>A configuration file to use with the GATK.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">String</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--gcs-max-retries">--gcs-max-retries </a>
			 / <small>-gcs-retries</small>
		</h3>
		<p class="args">
			<b>If the GCS bucket channel errors out, how many times it will attempt to re-initiate the connection</b><br />
			
		</p>
		<p>
		<span class="label label-info ">int</span>
			&nbsp;<span class="label">20</span>
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--gcs-project-for-requester-pays">--gcs-project-for-requester-pays </a>
			
		</h3>
		<p class="args">
			<b>Project to bill when accessing "requester pays" buckets. If unset, these buckets cannot be accessed.  User must have storage.buckets.get permission on the bucket being accessed.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">String</span>
			&nbsp;<span class="label">""</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--help">--help </a>
			 / <small>-h</small>
		</h3>
		<p class="args">
			<b>display the help message</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--ignore-all-filters">--ignore-all-filters </a>
			
		</h3>
		<p class="args">
			<b>If true, ignore all filters in the input VCF.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--ignore-filter">--ignore-filter </a>
			
		</h3>
		<p class="args">
			<b>Ignore the specified filter(s) in the input VCF.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">List[String]</span>
			&nbsp;<span class="label">[]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--input">--input </a>
			 / <small>-I</small>
		</h3>
		<p class="args">
			<b>BAM/SAM/CRAM file containing reads</b><br />
			
		</p>
		<p>
		<span class="label label-info ">List[GATKPath]</span>
			&nbsp;<span class="label">[]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--interval-exclusion-padding">--interval-exclusion-padding </a>
			 / <small>-ixp</small>
		</h3>
		<p class="args">
			<b>Amount of padding (in bp) to add to each interval you are excluding.</b><br />
			Use this to add padding to the intervals specified using -XL. For example, '-XL 1:100' with a
 padding value of 20 would turn into '-XL 1:80-120'. This is typically used to add padding around targets when
 analyzing exomes.
		</p>
		<p>
		<span class="label label-info ">int</span>
			&nbsp;<span class="label">0</span>
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--interval-merging-rule">--interval-merging-rule </a>
			 / <small>-imr</small>
		</h3>
		<p class="args">
			<b>Interval merging rule for abutting intervals</b><br />
			By default, the program merges abutting intervals (i.e. intervals that are directly side-by-side but do not
 actually overlap) into a single continuous interval. However you can change this behavior if you want them to be
 treated as separate intervals instead.
		</p>
			<p>
				The --interval-merging-rule argument is an enumerated type (IntervalMergingRule), which can have one of the following values:
			<dl class="enum">
					<dt class="enum">ALL</dt>
					<dd class="enum"></dd>
					<dt class="enum">OVERLAPPING_ONLY</dt>
					<dd class="enum"></dd>
			</dl>
			</p>
		<p>
		<span class="label label-info ">IntervalMergingRule</span>
			&nbsp;<span class="label">ALL</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--interval-padding">--interval-padding </a>
			 / <small>-ip</small>
		</h3>
		<p class="args">
			<b>Amount of padding (in bp) to add to each interval you are including.</b><br />
			Use this to add padding to the intervals specified using -L. For example, '-L 1:100' with a
 padding value of 20 would turn into '-L 1:80-120'. This is typically used to add padding around targets when
 analyzing exomes.
		</p>
		<p>
		<span class="label label-info ">int</span>
			&nbsp;<span class="label">0</span>
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--interval-set-rule">--interval-set-rule </a>
			 / <small>-isr</small>
		</h3>
		<p class="args">
			<b>Set merging approach to use for combining interval inputs</b><br />
			By default, the program will take the UNION of all intervals specified using -L and/or -XL. However, you can
 change this setting for -L, for example if you want to take the INTERSECTION of the sets instead. E.g. to
 perform the analysis only on chromosome 1 exomes, you could specify -L exomes.intervals -L 1 --interval-set-rule
 INTERSECTION. However, it is not possible to modify the merging approach for intervals passed using -XL (they will
 always be merged using UNION).

 Note that if you specify both -L and -XL, the -XL interval set will be subtracted from the -L interval set.
		</p>
			<p>
				The --interval-set-rule argument is an enumerated type (IntervalSetRule), which can have one of the following values:
			<dl class="enum">
					<dt class="enum">UNION</dt>
					<dd class="enum">Take the union of all intervals</dd>
					<dt class="enum">INTERSECTION</dt>
					<dd class="enum">Take the intersection of intervals (the subset that overlaps all intervals specified)</dd>
			</dl>
			</p>
		<p>
		<span class="label label-info ">IntervalSetRule</span>
			&nbsp;<span class="label">UNION</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--intervals">--intervals </a>
			 / <small>-L</small>
		</h3>
		<p class="args">
			<b>One or more genomic intervals over which to operate</b><br />
			
		</p>
		<p>
		<span class="label label-info ">List[String]</span>
			&nbsp;<span class="label">[]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--lenient">--lenient </a>
			 / <small>-LE</small>
		</h3>
		<p class="args">
			<b>Lenient processing of VCF files</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--max-variants-per-shard">--max-variants-per-shard </a>
			
		</h3>
		<p class="args">
			<b>If non-zero, partitions VCF output into shards, each containing up to the given number of records.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">int</span>
			&nbsp;<span class="label">0</span>
			&nbsp;<span class="label label-warning">[ [ 0</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--maximum-number-of-unlabeled-variants">--maximum-number-of-unlabeled-variants </a>
			
		</h3>
		<p class="args">
			<b>Maximum number of unlabeled variants to extract. If greater than zero, reservoir sampling will be used to randomly sample this number of sites from input sites that are not present in the specified resources. Choice of this number should be guided by considerations for training the negative model in TrainVariantAnnotationsModel; users may wish to choose a number that is comparable to the expected size of the labeled training set or that is compatible with available memory resources.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">int</span>
			&nbsp;<span class="label">0</span>
			&nbsp;<span class="label label-warning">[ [ 0</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--mode">--mode </a>
			
		</h3>
		<p class="args">
			<b>Variant types to extract.</b><br />
			
		</p>
			<p>
				The --mode argument is an enumerated type (List[VariantType]), which can have one of the following values:
			<dl class="enum">
					<dt class="enum">SNP</dt>
					<dd class="enum"></dd>
					<dt class="enum">INDEL</dt>
					<dd class="enum"></dd>
			</dl>
			</p>
		<p>
		<span class="label label-info ">List[VariantType]</span>
			&nbsp;<span class="label">[SNP, INDEL]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--omit-alleles-in-hdf5">--omit-alleles-in-hdf5 </a>
			
		</h3>
		<p class="args">
			<b>If true, omit alleles in output HDF5 files in order to decrease file sizes.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--output">--output </a>
			 / <small>-O</small>
		</h3>
		<p class="args">
			<b>Prefix for output filenames.</b><br />
			
		</p>
		<p>
				<span class="badge badge-important">R</span>
		<span class="label label-info ">String</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--QUIET">--QUIET </a>
			
		</h3>
		<p class="args">
			<b>Whether to suppress job-summary info on System.err.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">Boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--read-filter">--read-filter </a>
			 / <small>-RF</small>
		</h3>
		<p class="args">
			<b>Read filters to be applied before analysis</b><br />
			
		</p>
		<p>
		<span class="label label-info ">List[String]</span>
			&nbsp;<span class="label">[]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--read-index">--read-index </a>
			 / <small>-read-index</small>
		</h3>
		<p class="args">
			<b>Indices to use for the read inputs. If specified, an index must be provided for every read input and in the same order as the read inputs. If this argument is not specified, the path to the index for each input will be inferred automatically.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">List[GATKPath]</span>
			&nbsp;<span class="label">[]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--read-validation-stringency">--read-validation-stringency </a>
			 / <small>-VS</small>
		</h3>
		<p class="args">
			<b>Validation stringency for all SAM/BAM/CRAM/SRA files read by this program.  The default stringency value SILENT can improve performance when processing a BAM file in which variable-length data (read, qualities, tags) do not otherwise need to be decoded.</b><br />
			
		</p>
			<p>
				The --read-validation-stringency argument is an enumerated type (ValidationStringency), which can have one of the following values:
			<dl class="enum">
					<dt class="enum">STRICT</dt>
					<dd class="enum"></dd>
					<dt class="enum">LENIENT</dt>
					<dd class="enum"></dd>
					<dt class="enum">SILENT</dt>
					<dd class="enum"></dd>
			</dl>
			</p>
		<p>
		<span class="label label-info ">ValidationStringency</span>
			&nbsp;<span class="label">SILENT</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--reference">--reference </a>
			 / <small>-R</small>
		</h3>
		<p class="args">
			<b>Reference sequence</b><br />
			
		</p>
		<p>
		<span class="label label-info ">GATKPath</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--reservoir-sampling-random-seed">--reservoir-sampling-random-seed </a>
			
		</h3>
		<p class="args">
			<b>Random seed to use for reservoir sampling of unlabeled variants.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">int</span>
			&nbsp;<span class="label">0</span>
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--resource">--resource </a>
			
		</h3>
		<p class="args">
			<b>Resource VCFs used to label extracted variants.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">List[FeatureInput[VariantContext]]</span>
			&nbsp;<span class="label">[]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--resource-matching-strategy">--resource-matching-strategy </a>
			
		</h3>
		<p class="args">
			<b>The strategy to use for determining whether an input variant is present in a resource in non-allele-specific mode (--use-allele-specific-annotations false). START_POSITION: Start positions of input and resource variants must match. START_POSITION_AND_GIVEN_REPRESENTATION: The intersection of the sets of input and resource alleles (in their given representations) must also be non-empty. START_POSITION_AND_MINIMAL_REPRESENTATION: The intersection of the sets of input and resource alleles (after converting alleles to their minimal representations) must also be non-empty. This argument has no effect in allele-specific mode (--use-allele-specific-annotations true), in which the minimal representations of the input and resource alleles must match.</b><br />
			
		</p>
			<p>
				The --resource-matching-strategy argument is an enumerated type (ResourceMatchingStrategy), which can have one of the following values:
			<dl class="enum">
					<dt class="enum">START_POSITION</dt>
					<dd class="enum"></dd>
					<dt class="enum">START_POSITION_AND_GIVEN_REPRESENTATION</dt>
					<dd class="enum"></dd>
					<dt class="enum">START_POSITION_AND_MINIMAL_REPRESENTATION</dt>
					<dd class="enum"></dd>
			</dl>
			</p>
		<p>
		<span class="label label-info ">ResourceMatchingStrategy</span>
			&nbsp;<span class="label">START_POSITION</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--seconds-between-progress-updates">--seconds-between-progress-updates </a>
			 / <small>-seconds-between-progress-updates</small>
		</h3>
		<p class="args">
			<b>Output traversal statistics every time this many seconds elapse</b><br />
			
		</p>
		<p>
		<span class="label label-info ">double</span>
			&nbsp;<span class="label">10.0</span>
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--sequence-dictionary">--sequence-dictionary </a>
			 / <small>-sequence-dictionary</small>
		</h3>
		<p class="args">
			<b>Use the given sequence dictionary as the master/canonical sequence dictionary.  Must be a .dict file.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">GATKPath</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--showHidden">--showHidden </a>
			 / <small>-showHidden</small>
		</h3>
		<p class="args">
			<b>display hidden arguments</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--sites-only-vcf-output">--sites-only-vcf-output </a>
			
		</h3>
		<p class="args">
			<b>If true, don't emit genotype fields when writing vcf file output.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--tmp-dir">--tmp-dir </a>
			
		</h3>
		<p class="args">
			<b>Temp directory to use.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">GATKPath</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--use-allele-specific-annotations">--use-allele-specific-annotations </a>
			
		</h3>
		<p class="args">
			<b>If true, use the allele-specific versions of the specified annotations.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--use-jdk-deflater">--use-jdk-deflater </a>
			 / <small>-jdk-deflater</small>
		</h3>
		<p class="args">
			<b>Whether to use the JdkDeflater (as opposed to IntelDeflater)</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--use-jdk-inflater">--use-jdk-inflater </a>
			 / <small>-jdk-inflater</small>
		</h3>
		<p class="args">
			<b>Whether to use the JdkInflater (as opposed to IntelInflater)</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--variant">--variant </a>
			 / <small>-V</small>
		</h3>
		<p class="args">
			<b>A VCF file containing variants</b><br />
			
		</p>
		<p>
				<span class="badge badge-important">R</span>
		<span class="label label-info ">GATKPath</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--verbosity">--verbosity </a>
			 / <small>-verbosity</small>
		</h3>
		<p class="args">
			<b>Control verbosity of logging.</b><br />
			
		</p>
			<p>
				The --verbosity argument is an enumerated type (LogLevel), which can have one of the following values:
			<dl class="enum">
					<dt class="enum">ERROR</dt>
					<dd class="enum"></dd>
					<dt class="enum">WARNING</dt>
					<dd class="enum"></dd>
					<dt class="enum">INFO</dt>
					<dd class="enum"></dd>
					<dt class="enum">DEBUG</dt>
					<dd class="enum"></dd>
			</dl>
			</p>
		<p>
		<span class="label label-info ">LogLevel</span>
			&nbsp;<span class="label">INFO</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--version">--version </a>
			
		</h3>
		<p class="args">
			<b>display the version number for this tool</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>

        <hr>
        <p><a href='#top'><i class='fa fa-chevron-up'></i> Return to top</a></p>
        <hr>
        <p class="see-also">See also 
        	<a href="https://software.broadinstitute.org/gatk/documentation/">General Documentation</a> |
        	<a class="hide_me_php" href="index.html">Tool Docs Index</a> <a class="hide_me_html" href="index">Tool Documentation Index</a> |
        	<a href="https://gatkforums.broadinstitute.org/gatk/">Support Forum</a>
        </p>

        <p class="version">GATK version 4.3.0.0 built at Wed, 12 Oct 2022 21:04:44 -0400.
        </p>

</div></div>

	<?php printFooter($module); ?>